--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Northern_Rebirth/scenarios/05a_01_The_Pursuit.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Northern_Rebirth/scenarios/05a_01_The_Pursuit.cfg	2020-08-25 00:28:42.000000000 -0400
@@ -26,8 +26,9 @@
     [side]
         side=1
         controller=human
-        recruit=Peasant,Woodsman,Thug,Poacher,Footpad
-        {GOLD 600 400 300}
+        recruit=Peasant,Woodsman,Thug,Poacher,Footpad,Ruffian
+        {GOLD 650 400 300}
+        {INCOME 3 2 1}
         team_name=rebels
         user_team_name= _ "Alliance"
 
@@ -104,7 +105,7 @@
 
         gold=0
         # {GOLD 100 150 250}
-        {INCOME 10 16 22}
+        {INCOME 6 16 22}
         team_name=undead
         user_team_name= _ "Undead"
         {FLAG_VARIANT undead}
@@ -510,7 +511,7 @@
 
         [gold]
             side=4
-            amount={ON_DIFFICULTY 200 350 500}
+            amount={ON_DIFFICULTY 120 350 500}
         [/gold]
     [/event]
 
@@ -1210,10 +1211,10 @@
             message= _ "Bring it on!"
         [/message]
 
-        # Add 200, 350 or 500 gold to treasury guards.
+        # Add 120, 350 or 500 gold to treasury guards.
         [gold]
             side=5
-            amount={ON_DIFFICULTY 200 350 500}
+            amount={ON_DIFFICULTY 120 350 500}
         [/gold]
     [/event]
 
@@ -1544,7 +1545,7 @@
 
         [gold]
             side=7
-            amount={ON_DIFFICULTY 200 350 500}
+            amount={ON_DIFFICULTY 120 350 500}
         [/gold]
     [/event]
 
@@ -1865,7 +1866,7 @@
 
         [gold]
             side=6
-            amount={ON_DIFFICULTY 200 350 500}
+            amount={ON_DIFFICULTY 120 350 500}
         [/gold]
     [/event]
 
@@ -1916,7 +1917,7 @@
 
         [gold]
             side=3
-            amount={ON_DIFFICULTY 200 250 300}
+            amount={ON_DIFFICULTY 120 250 300}
         [/gold]
     [/event]
 
