--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Northern_Rebirth/scenarios/07a_Settling_Disputes.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Northern_Rebirth/scenarios/07a_Settling_Disputes.cfg	2020-08-15 19:42:57.000000000 -0400
@@ -125,7 +125,9 @@
         side=1
         controller=human
         recruit=Thug,Poacher,Spearman,Bowman,Footpad
-        gold=0 #player has large carryover here, is handled in prestart event
+        # player is supposed to have a large carryover here (handled in prestart event), but just in case:
+        {GOLD 20 10 0}
+        {INCOME 2 1 0}
         team_name=knalgans
         user_team_name= _ "Knalgans"
 
@@ -143,7 +145,7 @@
         side=2
         controller=ai
         recruit=Dwarvish Ulfserker
-        {GOLD 100 90 60}
+        {GOLD 120 90 60}
         team_name=knalgans
         user_team_name= _ "Knalgans"
         {FLAG_VARIANT knalgan}
@@ -158,7 +160,7 @@
             {TRAIT_HEALTHY}
         [/modifications]
 
-        hitpoints=1
+        {QUANTITY hitpoints 3 2 1}
         fog=yes
         share_view=yes
         facing=se
@@ -166,43 +168,43 @@
         # Place some combatants at various states of hp/xp
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 28 8}
         [+unit]
-            hitpoints=12
-            experience=40
+            {QUANTITY hitpoints 16 14 12}
+            {QUANTITY experience 41 40 39}
             facing=se
         [/unit]
 
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 24 7}
         [+unit]
-            hitpoints=34
-            experience=8
+            hitpoints=34 # max
+            {QUANTITY experience 10 9 8}
             facing=ne
         [/unit]
 
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 24 4}
         [+unit]
-            hitpoints=28
-            experience=18
+            {QUANTITY hitpoints 32 30 28}
+            {QUANTITY experience 22 20 18}
             facing=sw
         [/unit]
 
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 23 2}
         [+unit]
-            hitpoints=30
-            experience=16
+            {QUANTITY hitpoints 34 32 30}
+            {QUANTITY experience 20 18 16}
             facing=s
         [/unit]
 
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 30 7}
         [+unit]
-            hitpoints=12
-            experience=16
+            {QUANTITY hitpoints 16 14 12}
+            {QUANTITY experience 20 18 16}
             facing=nw
         [/unit]
 
         {NOTRAIT_UNIT 2 "Dwarvish Ulfserker" 22 6}
         [+unit]
-            hitpoints=9
-            experience=41
+            {QUANTITY hitpoints 15 12 9}
+            experience=41 # 42 is max
             facing=sw
         [/unit]
     [/side]
@@ -212,7 +214,7 @@
         side=3
         controller=ai
         recruit=Gryphon,Young Ogre,Ogre,Troll Whelp,Troll,Troll Rocklobber,Red Mage,Mage
-        {GOLD 950 1100 1600}
+        {GOLD 900 1100 1600}
         team_name=liches
         user_team_name= _ "Liches"
         {FLAG_VARIANT undead}
@@ -227,29 +229,29 @@
 
         {NOTRAIT_UNIT 3 Troll 30 9}
         [+unit]
-            hitpoints=36
-            experience=14
+            {QUANTITY hitpoints 32 34 36}
+            {QUANTITY experience 10 12 14}
             facing=nw
         [/unit]
 
         {NOTRAIT_UNIT 3 "Young Ogre" 28 6}
         [+unit]
-            hitpoints=6
-            experience=31
+            {QUANTITY hitpoints 4 5 6}
+            {QUANTITY experience 29 30 31}
             facing=se
         [/unit]
 
         {NOTRAIT_UNIT 3 "Young Ogre" 20 7}
         [+unit]
-            hitpoints=12
-            experience=12
+            {QUANTITY hitpoints 10 11 12}
+            {QUANTITY experience 10 11 12}
             facing=ne
         [/unit]
 
         {NOTRAIT_UNIT 3 "Ogre" 23 3}
         [+unit]
-            hitpoints=27
-            experience=8
+            {QUANTITY hitpoints 21 24 27}
+            {QUANTITY experience 6 7 8}
             facing=n
         [/unit]
 
@@ -289,7 +291,7 @@
         [/store_gold]
         [modify_side]
             side=1
-            {GOLD 200 100 80}
+            {GOLD 280 180 80}
         [/modify_side]
 
         # Recall heroes
