--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Northern_Rebirth/scenarios/08a_Elvish_Princess.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Northern_Rebirth/scenarios/08a_Elvish_Princess.cfg	2020-08-23 00:34:39.000000000 -0400
@@ -4,7 +4,7 @@
     id=08a_Elvish_Princess
     name= _ "Elvish Princess"
     map_data="{campaigns/Northern_Rebirth/maps/08a_Elvish_Princess.map}"
-    {TURNS 21 18 15}
+    {TURNS 23 19 15}
     next_scenario=09a_Introductions
     victory_when_enemies_defeated=no
 
@@ -94,9 +94,12 @@
     [side]
         side=1
         controller=human
-        recruit=Thug,Poacher,Spearman,Bowman,Gryphon,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Footpad
+        # Tallin only asked the mages for their gryphons at the end of the previous scenario, so don't add any more of their units here:
+        recruit=Thug,Poacher,Spearman,Bowman,Gryphon,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Footpad,Fencer,Heavy Infantryman
         team_name=knalgans
         user_team_name= _ "Knalgans"
+        {GOLD 99 66 33}
+        {INCOME 9 6 3}
 
         # wmllint: recognize Tallin
         {CHARACTER_STATS_TALLIN}
@@ -107,8 +110,8 @@
         side=2
         controller=ai
         recruit=Orcish Grunt,Orcish Archer,Wolf Rider,Orcish Assassin,Troll Whelp
-        {GOLD 300 400 500}
-        {INCOME 15 20 25}
+        {GOLD 250 375 500}
+        {INCOME 10 20 25}
         team_name=Orcs
         user_team_name= _ "Orcs"
         {FLAG_VARIANT6 ragged}
@@ -166,8 +169,10 @@
         {GENERIC_UNIT 3 "Orcish Crossbowman" 17 14} {GUARDIAN}
     [/side]
 
-    {STARTING_VILLAGES 1 5}
-    {STARTING_VILLAGES_AREA 2 15 17 12}
+    # 1 = Tallin, so get smaller by difficulty:
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 7 6 5}}
+    # 2 = Orcs, so get bigger by difficulty:
+    {STARTING_VILLAGES_AREA 2 15 17 {ON_DIFFICULTY 10 11 12}}
 
     {PLACE_PRISONER_IMAGE units/elves-wood/sorceress.png 9 14}
 
@@ -362,6 +367,26 @@
         [/message]
     [/event]
 
+#ifdef EASY
+    [event]
+        name=last breath
+        [filter]
+            id=Atul
+        [/filter]
+        [message]
+            race=human
+            message= _ "This orc was carrying some gold!"
+        [/message]
+        [sound]
+            name=gold.ogg
+        [/sound]
+        [gold]
+            side=1
+            {QUANTITY amount 120 80 40}
+        [/gold]
+    [/event]
+#endif
+
     # Freeing Eryssa event - victory proceed to introductions
     [event]
         name=moveto
