--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Northern_Rebirth/scenarios/09a_Introductions.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Northern_Rebirth/scenarios/09a_Introductions.cfg	2020-08-23 01:30:43.000000000 -0400
@@ -20,7 +20,7 @@
     [side]
         side=1
         controller=human
-        recruit=Thug,Poacher,Spearman,Bowman,Gryphon,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Footpad
+        recruit=Thug,Poacher,Spearman,Bowman,Gryphon,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Footpad,Fencer,Heavy Infantryman
         team_name=knalgans
         user_team_name=_ "Knalgans"
 
@@ -44,7 +44,7 @@
         {GENERIC_UNIT 2 "Troll" 10 6}
         {GENERIC_UNIT 2 "Troll" 11 4}
         {GENERIC_UNIT 2 "Troll" 16 6}
-        {GENERIC_UNIT 2 "Troll" 18 11}
+        {GENERIC_UNIT 2 {ON_DIFFICULTY "Troll Whelp" "Troll" "Troll Warrior"} 18 11}
 
         {GENERIC_UNIT 2 "Troll Whelp" 3 9}
         {GENERIC_UNIT 2 "Troll Whelp" 6 3}
@@ -58,7 +58,7 @@
         {GENERIC_UNIT 2 "Troll Rocklobber" 4 3}
         {GENERIC_UNIT 2 "Troll Rocklobber" 9 3}
         {GENERIC_UNIT 2 "Troll Rocklobber" 10 5}
-        {GENERIC_UNIT 2 "Troll Rocklobber" 14 3}
+        {GENERIC_UNIT 2 {ON_DIFFICULTY "Troll Whelp" "Troll Rocklobber" "Troll Rocklobber"} 14 3}
     [/side]
 
     {PLACE_IMAGE scenery/signpost.png 1 1}
@@ -103,6 +103,10 @@
         [/recall]
 
         [recall]
+            id=Varem
+        [/recall]
+
+        [recall]
             id="Ro'Arthian"
         [/recall]
 
