--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Northern_Rebirth/scenarios/13a_Showdown.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Northern_Rebirth/scenarios/13a_Showdown.cfg	2020-10-09 02:07:02.000000000 -0400
@@ -41,11 +41,13 @@
         canrecruit=yes
         side=1
         controller=human
-        recruit=Thug,Poacher,Spearman,Bowman,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Gryphon,Footpad
+        recruit=Thug,Poacher,Spearman,Bowman,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Dwarvish Scout,Gryphon,Footpad,Fencer,Heavy Infantryman
         team_name=knalgans
         user_team_name=_"Alliance"
-        gold=100
+        {GOLD 900 500 100}
+        {INCOME 18 12 9}
     [/side]
+    # This gets all the ones nearby; keep constant:
     {STARTING_VILLAGES 1 5}
     # Main enemy
     [side]
@@ -63,39 +65,55 @@
                     x=30
                     y=30
                 [/criteria]
-                value=400
+                value={ON_DIFFICULTY 200 300 400}
+            [/goal]
+            [goal]
+                name=protect_unit
+                [criteria]
+                    id=Rakshas
+                [/criteria]
+                protect_radius={ON_DIFFICULTY 12 11 10}
+                value={ON_DIFFICULTY 250 350 450}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=1
+                    side=1 # Tallin
                 [/criteria]
-                value=300
+                value={ON_DIFFICULTY 60 180 300}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=7
+                    side=7 # Krash, or Ro'Arthian
                 [/criteria]
-                value=150
+                value={ON_DIFFICULTY 50 100 150}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=8
+                    side=8 # Hamel
                 [/criteria]
-                value=50
+                value={ON_DIFFICULTY 40 50 60}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=9
+                    id=Hamel
                 [/criteria]
-                value=250
+                value={ON_DIFFICULTY 30 40 50}
+            [/goal]
+            [goal]
+                name=target
+                [criteria]
+                    side=9 # Eryssa, or Ro'Sothian
+                [/criteria]
+                value={ON_DIFFICULTY 100 175 250}
             [/goal]
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 2 3}
     # Sidekicks
     [side]
@@ -111,36 +129,52 @@
         user_team_name=_"Orcs"
         [ai]
             [goal]
+                name=protect_unit
+                [criteria]
+                    id="Ha'Tuil"
+                [/criteria]
+                protect_radius={ON_DIFFICULTY 9 8 7}
+                value={ON_DIFFICULTY 250 350 450}
+            [/goal]
+            [goal]
                 name=target
                 [criteria]
-                    side=1
+                    side=1 # Tallin
                 [/criteria]
-                value=400
+                value={ON_DIFFICULTY 160 280 400}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=7
+                    side=7 # Krash, or Ro'Arthian
                 [/criteria]
-                value=300
+                value={ON_DIFFICULTY 100 200 300}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=8
+                    side=8 # Hamel
                 [/criteria]
-                value=200
+                value={ON_DIFFICULTY 80 160 240}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=9
+                    id=Hamel
+                [/criteria]
+                value={ON_DIFFICULTY 90 180 270}
+            [/goal]
+            [goal]
+                name=target
+                [criteria]
+                    side=9 # Eryssa, or Ro'Sothian
                 [/criteria]
-                value=350
+                value={ON_DIFFICULTY 150 250 350}
             [/goal]
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 3 4}
 
     [side]
@@ -156,36 +190,52 @@
         user_team_name=_"Orcs"
         [ai]
             [goal]
+                name=protect_unit
+                [criteria]
+                    id="Ar'Muff"
+                [/criteria]
+                protect_radius={ON_DIFFICULTY 9 8 7}
+                value={ON_DIFFICULTY 250 350 450}
+            [/goal]
+            [goal]
                 name=target
                 [criteria]
-                    side=1
+                    side=1 # Tallin
                 [/criteria]
-                value=400
+                value={ON_DIFFICULTY 160 280 400}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=7
+                    side=7 # Krash, or Ro'Arthian
                 [/criteria]
-                value=300
+                value={ON_DIFFICULTY 100 200 300}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=8
+                    side=8 # Hamel
                 [/criteria]
-                value=200
+                value={ON_DIFFICULTY 80 160 240}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=9
+                    id=Hamel
+                [/criteria]
+                value={ON_DIFFICULTY 90 180 270}
+            [/goal]
+            [goal]
+                name=target
+                [criteria]
+                    side=9 # Eryssa, or Ro'Sothian
                 [/criteria]
-                value=350
+                value={ON_DIFFICULTY 150 250 350}
             [/goal]
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 4 4}
 
     [side]
@@ -201,36 +251,52 @@
         user_team_name=_"Orcs"
         [ai]
             [goal]
+                name=protect_unit
+                [criteria]
+                    id="Calter"
+                [/criteria]
+                protect_radius={ON_DIFFICULTY 9 8 7}
+                value={ON_DIFFICULTY 250 350 450}
+            [/goal]
+            [goal]
                 name=target
                 [criteria]
-                    side=1
+                    side=1 # Tallin
                 [/criteria]
-                value=400
+                value={ON_DIFFICULTY 160 280 400}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=7
+                    side=7 # Krash, or Ro'Arthian
                 [/criteria]
-                value=300
+                value={ON_DIFFICULTY 100 200 300}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=8
+                    side=8 # Hamel
                 [/criteria]
-                value=200
+                value={ON_DIFFICULTY 80 160 240}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=9
+                    id=Hamel
+                [/criteria]
+                value={ON_DIFFICULTY 90 180 270}
+            [/goal]
+            [goal]
+                name=target
+                [criteria]
+                    side=9 # Eryssa, or Ro'Sothian
                 [/criteria]
-                value=350
+                value={ON_DIFFICULTY 150 250 350}
             [/goal]
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 5 4}
 
     [side]
@@ -246,36 +312,52 @@
         user_team_name=_"Orcs"
         [ai]
             [goal]
+                name=protect_unit
+                [criteria]
+                    id="Halter"
+                [/criteria]
+                protect_radius={ON_DIFFICULTY 9 8 7}
+                value={ON_DIFFICULTY 250 350 450}
+            [/goal]
+            [goal]
                 name=target
                 [criteria]
-                    side=1
+                    side=1 # Tallin
                 [/criteria]
-                value=400
+                value={ON_DIFFICULTY 160 280 400}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=7
+                    side=7 # Krash, or Ro'Arthian
                 [/criteria]
-                value=300
+                value={ON_DIFFICULTY 100 200 300}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=8
+                    side=8 # Hamel
                 [/criteria]
-                value=200
+                value={ON_DIFFICULTY 80 160 240}
             [/goal]
             [goal]
                 name=target
                 [criteria]
-                    side=9
+                    id=Hamel
+                [/criteria]
+                value={ON_DIFFICULTY 90 180 270}
+            [/goal]
+            [goal]
+                name=target
+                [criteria]
+                    side=9 # Eryssa, or Ro'Sothian
                 [/criteria]
-                value=350
+                value={ON_DIFFICULTY 150 250 350}
             [/goal]
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 6 4}
 
     # Krash may not actually still be alive at this point, but setting up
@@ -285,7 +367,8 @@
         side=7
         save_id=Krash
         side_name= _ "Krash"
-        gold=200
+        {GOLD 600 400 200}
+        {INCOME 9 6 3}
         recruit=Drake Fighter,Drake Glider,Drake Burner,Drake Clasher
         team_name=knalgans
         user_team_name=_"Alliance"
@@ -353,6 +436,7 @@
         [/ai]
         {FLAG_VARIANT knalgan}
     [/side]
+    # Gets all ones nearby; keep constant:
     {STARTING_VILLAGES 8 10}
 
     {AI_CONTROLLER () 1 8 ()}
@@ -360,7 +444,8 @@
     # Side for Eryssa if alive
     [side]
         side=9
-        gold=200
+        {GOLD 600 400 200}
+        {INCOME 9 6 3}
         recruit=Elvish Fighter,Elvish Archer,Elvish Shaman,Elvish Scout,Elvish Captain,Elvish Hero,Elvish Ranger,Elvish Marksman,Elvish Rider,Elvish Druid,Elvish Sorceress
         team_name=knalgans
         user_team_name=_"Alliance"
@@ -374,9 +459,62 @@
     # STARTING_VILLAGES works due to defeat_condition & having a starting location
     {STARTING_VILLAGES 9 10}
 
+    # This should help prevent Hamel from suiciding himself by discouraging him
+    # from leaving his castle and surrounding villages
+    # (copied from SotBE)
+    [event]
+        name=side 8 turn refresh
+        first_time_only=no
+
+        [modify_unit]
+            [filter]
+                side=$side_number
+                canrecruit=yes
+            [/filter]
+
+            [movement_costs]
+                flat={ON_DIFFICULTY 99 66 33}
+            [/movement_costs]
+        [/modify_unit]
+
+        [event]
+            name=side turn end
+
+            [modify_unit]
+                [filter]
+                    side=$side_number
+                    canrecruit=yes
+                [/filter]
+
+                [movement_costs]
+                    flat=1
+                [/movement_costs]
+            [/modify_unit]
+        [/event]
+    [/event]
+
     [event]
         name=prestart
 
+#ifdef EASY
+        [terrain]
+            x,y=54,30
+            terrain=Gg^Vh
+        [/terrain]
+        [terrain]
+            x,y=54,31
+            terrain=Gg^Vh
+        [/terrain]
+        [capture_village]
+            side=1
+            x,y=54,30
+        [/capture_village]
+        [capture_village]
+            side=1
+            x,y=54,31
+        [/capture_village]
+#endif
+
         # Check if Krash is alive, if yes place him at helm of side 7,
         # otherwise it's taken over by older lich.
         {RECALL_KRASH 7}
@@ -417,10 +555,14 @@
                     side=7
                     type=Troll,Troll Whelp,Ogre,Young Ogre,Mage,Red Mage,Gryphon
                 [/allow_recruit]
-                # dwarves also get gold if no krash, since it's much harder to defend them
                 [gold]
-                    side=7,8
-                    amount=400
+                    side=7
+                    amount={ON_DIFFICULTY 650 525 400}
+                [/gold]
+                # dwarves also get gold if no Krash, since it's harder to defend them
+                [gold]
+                    side=8
+                    amount={ON_DIFFICULTY 420 410 400}
                 [/gold]
                 [modify_side]
                     side=7
@@ -495,8 +637,13 @@
                     type=Troll,Troll Whelp,Ogre,Young Ogre,Mage,Red Mage,Gryphon
                 [/allow_recruit]
                 [gold]
-                    side=8,9
-                    amount=400
+                    side=9
+                    amount={ON_DIFFICULTY 650 525 400}
+                [/gold]
+                # dwarves also get gold if no Eryssa, since it's harder to defend them
+                [gold]
+                    side=8
+                    amount={ON_DIFFICULTY 420 410 400}
                 [/gold]
                 [modify_side]
                     side=9
@@ -927,6 +1074,7 @@
             message= _ "FOR HIDEL!!"
         [/message]
         [message]
+            # TODO: Make sure Supporter isn't one of the characters who already said something:
             role=Supporter
             message= _ "FOR DWARVEN DOORS!!"
         [/message]
@@ -987,7 +1135,7 @@
             message= _ "Yeah. Obviously this is going to take some time. If necessary we can sit tight in our fortifications and besiege them until they starve!"
         [/message]
     [/event]
-    # Rakashas death
+    # Rakshas death
     [event]
         name=last breath
         [filter]
