--- /Users/ericgallager/Downloads/Flight_Freedom/scenarios/13_Betrayal.cfg	2019-02-28 17:47:53.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/Flight_Freedom/scenarios/13_Betrayal.cfg	2020-10-05 20:06:46.000000000 -0400
@@ -3,8 +3,9 @@
 [scenario]
     name= _ "Betrayal"
     {MAP 13}
-    turns=30
+    {TURNS 32 31 30}
     {SCENARIO_MUSIC frantic.ogg}
+    
     id=Betrayal
     next_scenario="Hordes_Undead"
 
@@ -27,7 +28,8 @@
         controller=human
         recruit=Drake Hatchling,Drake Slave,Drake Burner,Drake Glider,Drake Fighter,Drake Clasher
         unrenamable=yes
-        gold=200
+        # explicit gold is probably unnecessary due to carryover from River of Skulls, but just in case:
+        {GOLD 220 210 200}
         team_name=good
         user_team_name= _ "Kakatar Tribe"
         fog=no
@@ -80,12 +82,18 @@
         name=start
         [lock_view]
         [/lock_view]
-		[delay]
-			time=750
-		[/delay]
+        [delay]
+            time=750
+        [/delay]
         #cave collapses
         {QUAKE cave-in.ogg}
+        [delay]
+            time=100
+        [/delay]
         {QUAKE_NOSOUND}
+        [delay]
+            time=50
+        [/delay]
         [terrain]
             x=9-12
             y=0-3
@@ -99,6 +107,7 @@
             side=1
             variable=gold
         [/store_gold]
+        # this percentage can stay constant; varying it would be overkill:
         {VARIABLE_OP gold multiply 0.3}
         [set_variable]
             name=gold
@@ -156,13 +165,13 @@
         [/message]
     [/event]
 
-	[event]
-		name=turn 26
-		[message]
-			speaker=Malakar
-			message= _ "Every minute we fight each other, is another step closer our pursuers approach! We must end this quickly!"
-		[/message]
-	[/event]
+    [event]
+        name=turn 26
+        [message]
+            speaker=Malakar
+            message= _ "Every minute we fight each other, is another step closer our pursuers approach! We must end this quickly!"
+        [/message]
+    [/event]
 
     [event]
         name=attack
@@ -200,10 +209,10 @@
         name=attack
         [filter]
             side=1
-			#he's still a pirate, justice probably isn't his highest motivation
-			[not]
-				id=Kogw
-			[/not]
+            #he's still a pirate, justice probably isn't his highest motivation
+            [not]
+                id=Kogw
+            [/not]
         [/filter]
         [filter_second]
             id=Theracar
