--- /Users/ericgallager/Downloads/Flight_Freedom/scenarios/16_Exodus.cfg	2019-02-28 17:47:53.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/Flight_Freedom/scenarios/16_Exodus.cfg	2020-10-05 22:51:12.000000000 -0400
@@ -3,7 +3,7 @@
 [scenario]
     name= _ "The Exodus"
     {MAP 16}
-    {TURNS 50 33 32}
+    {TURNS 50 41 32}
     {SCENARIO_MUSIC battle.ogg}
     id=Exodus
     next_scenario="Blockade"
@@ -30,6 +30,8 @@
         team_name=good
         user_team_name= _ "Kakatar Tribe"
         fog=no
+        {GOLD 180 140 100}
+        {INCOME 3 2 1}
     [/side]
     [side]
         type=Highwayman
@@ -107,6 +109,7 @@
             unrenamable=yes
             {IS_HERO}
             random_traits=no
+            facing=se
             [modifications]
                 {TRAIT_LOYAL}
                 {TRAIT_STRONG}
@@ -133,6 +136,7 @@
             unrenamable=yes
             {IS_HERO}
             random_traits=no
+            facing=n
             [modifications]
                 {TRAIT_LOYAL}
                 {TRAIT_FEARLESS}
@@ -156,17 +160,19 @@
             speaker=Kogw
             message= _ "You old scoundrel! How in blazes are you still alive?"
         [/message]
-        #[store_unit]
-        #    [filter]
-        #        id=Rhys
-        #    [/filter]
-        #    variable=rhys_store
-        #[/store_unit]
-        #{VARIABLE rhys_store.name (_ "Rhys")}
-        #[unstore_unit]
-        #    variable=rhys_store
-        #[/unstore_unit]
-        #{CLEAR_VARIABLE rhys_store}
+#ifdef __UNUSED__
+        [store_unit]
+            [filter]
+                id=Rhys
+            [/filter]
+            variable=rhys_store
+        [/store_unit]
+        {VARIABLE rhys_store.name (_ "Rhys")}
+        [unstore_unit]
+            variable=rhys_store
+        [/unstore_unit]
+        {CLEAR_VARIABLE rhys_store}
+#endif
         [message]
             speaker=Rhys
             message= _ "You think a few measly warships could really stop the Unforgiver? We played 'possum until they were fully docked, then high-tailed it back North."
@@ -211,28 +217,26 @@
             speaker=Davies
             message= _ "Oh, and we've scraped together a few coins. We kind of figured we'd be using them to buy our way out of here, but as long as you're alive I suppose they're yours."
         [/message]
-#ifdef EASY
-        [gold]
-            side=1
-            amount=60
-        [/gold]
-#endif
-#ifdef NORMAL
-        [gold]
-            side=1
-            amount=35
-        [/gold]
-#endif
-#ifdef HARD
+
         [gold]
             side=1
-            amount=25
+            amount={ON_DIFFICULTY 60 45 30}
         [/gold]
-#endif
+
         [sound]
             name=gold.ogg
         [/sound]
 
+        [message]
+            speaker=Rhys
+            message= _ "Also, the village we were staying in is friendly."
+        [/message]
+
+        [capture_village]
+            side=1
+            x,y=32,46
+        [/capture_village]
+
         [unstore_unit]
             variable=unforgiver_store
             x,y=2,1
