--- /Users/ericgallager/Downloads/Flight_Freedom/scenarios/17_Blockade.cfg	2019-02-28 17:47:53.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/Flight_Freedom/scenarios/17_Blockade.cfg	2020-10-06 00:16:39.000000000 -0400
@@ -3,7 +3,7 @@
 [scenario]
     name= _ "Blockade"
     {MAP 17}
-    turns=30
+    {TURNS 36 33 30}
     {SCENARIO_MUSIC revelation.ogg}
     id=Blockade
     next_scenario="Return_Morogor"
@@ -30,6 +30,8 @@
         team_name=good
         user_team_name= _ "Skullbone Pirates"
         fog=no
+        {GOLD 180 140 100}
+        {INCOME 3 2 1}
     [/side]
     [side]
         type=Naga Myrmidon
@@ -39,22 +41,22 @@
         controller=ai
         canrecruit=yes
         {GOLD 400 450 500}
-        income=15
+        {INCOME 9 12 15}
         recruit=Naga Fighter,Naga Guardian,Naga Hunter,Naga Mage
         team_name=naga
         user_team_name= _ "Nagas"
         [ai]
             [goal]
                 [criteria]
-                    side=4
+                    side=4 # merfolk
                 [/criteria]
-                value=10
+                value={ON_DIFFICULTY 12 11 10}
             [/goal]
             [goal]
                 [criteria]
-                    side=1
+                    side=1 # you
                 [/criteria]
-                value=0.6
+                value={ON_DIFFICULTY 0.4 0.5 0.6}
             [/goal]
         [/ai]
     [/side]
@@ -79,9 +81,9 @@
             [/avoid]
             [goal]
                 [criteria]
-                    side=1
+                    side=1 # you
                 [/criteria]
-                value=15
+                value={ON_DIFFICULTY 9 12 15}
             [/goal]
         [/ai]
     [/side]
@@ -93,7 +95,7 @@
         controller=ai
         canrecruit=yes
         {GOLD 400 450 500}
-        income=20
+        {INCOME 16 18 20}
         recruit=Merman Fighter,Mermaid Initiate,Merman Hunter
         team_name=mermen
         user_team_name= _ "Mermen"
@@ -102,15 +104,15 @@
         [ai]
             [goal]
                 [criteria]
-                    side=2
+                    side=2 # nagas
                 [/criteria]
-                value=10
+                value={ON_DIFFICULTY 12 11 10}
             [/goal]
             [goal]
                 [criteria]
-                    side=1
+                    side=1 # you
                 [/criteria]
-                value=0.7
+                value={ON_DIFFICULTY 0.5 0.6 0.7}
             [/goal]
         [/ai]
     [/side]
@@ -161,21 +163,23 @@
         #just in case I need them later
         #actually, no - I think I'll let the player recall them after all
         #wmllint: recognize Rhys
-        #        [store_unit]
-        #            [filter]
-        #                id=Rhys
-        #            [/filter]
-        #            variable=rhys_store
-        #            kill=yes
-        #        [/store_unit]
-        #        #wmllint: recognize Davies
-        #        [store_unit]
-        #            [filter]
-        #                id=Davies
-        #            [/filter]
-        #            variable=davies_store
-        #            kill=yes
-        #        [/store_unit]
+#ifdef __UNUSED__
+        [store_unit]
+            [filter]
+                id=Rhys
+            [/filter]
+            variable=rhys_store
+            kill=yes
+        [/store_unit]
+        #wmllint: recognize Davies
+        [store_unit]
+            [filter]
+                id=Davies
+            [/filter]
+            variable=davies_store
+            kill=yes
+        [/store_unit]
+#endif
         [unit]
             side=2
             x,y=9,8
@@ -263,10 +267,13 @@
         {DSTALKER 8 6}
         {DSTALKER 4 7}
 #ifdef EASY
+        {PLACE_IMAGE scenery/shipwreck-1.png 5 6} 
 #else
         {DSTALKER 5 6}
 #endif
-        #{DSTALKER 9 7}
+#ifdef __UNUSED__
+        {DSTALKER 9 7}
+#endif
 #ifdef HARD
         {DSTALKER 2 6}
 #endif
@@ -278,9 +285,9 @@
     {STARTING_VILLAGES 4 10}
 
 #define RECRUIT_RUN_BLOCKADE X1 Y1 X2 Y2
-	{RECRUIT_RUN_FILTER {X1} {Y1} {X2} {Y2} 1 (
-	race=boat
-	)}
+    {RECRUIT_RUN_FILTER {X1} {Y1} {X2} {Y2} 1 (
+        race=boat
+    )}
 #enddef
 
     {RECRUIT_RUN_BLOCKADE 1 19 7 16}
@@ -421,10 +428,10 @@
 
     #so the mermen last a bit longer
     [event]
-        name=turn 13
+        name=turn 12
         [gold]
             side=4
-            amount=80
+            amount={ON_DIFFICULTY 60 70 80}
         [/gold]
     [/event]
 
@@ -464,19 +471,19 @@
         [/if]
     [/event]
 
-	[event]
-		name=enemies defeated
-		[message]
-			speaker=Unforgiver
-			image=$malakar_image
-			message= _ "(Malakar) The sea dwellers have been defeated!"
-		[/message]
-		[message]
-			speaker=Unforgiver
-			image=$kogw_image
-			message= _ "(Kogw) They've only been driven back. Quickly, we need to leave these waters! And be wary for any stragglers."
-		[/message]
-	[/event]
+    [event]
+        name=enemies defeated
+        [message]
+            speaker=Unforgiver
+            image=$malakar_image
+            message= _ "(Malakar) The sea dwellers have been defeated!"
+        [/message]
+        [message]
+            speaker=Unforgiver
+            image=$kogw_image
+            message= _ "(Kogw) They've only been driven back. Quickly, we need to leave these waters! And be wary for any stragglers."
+        [/message]
+    [/event]
 
     [event]
         name=victory
