--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg	2020-09-06 20:56:30.000000000 -0400
+++ campaigns/Eastern_Invasion/12_Evacuation.cfg	2020-08-30 20:35:55.000000000 -0400
@@ -3,7 +3,7 @@
     id=12_Evacuation
     name= _ "Evacuation"
     map_data="{campaigns/Eastern_Invasion/maps/12_Evacuation.map}"
-    turns=12
+    {TURNS 16 14 12}
     next_scenario=13_The_Drowned_Plains
 
     {DEFAULT_SCHEDULE}
@@ -30,7 +30,8 @@
         controller=human
         team_name=wesnothians
         user_team_name=_"Wesnothians"
-        gold=200
+        {GOLD 294 247 200}
+        {INCOME 9 6 3}
         {FLAG_VARIANT loyalist}
     [/side]
 
@@ -44,7 +45,7 @@
         controller=ai
         recruit=Orcish Warrior,Goblin Knight,Direwolf Rider
         {GOLD 100 175 225}
-        {INCOME 20 25 30}
+        {INCOME 10 20 30}
         team_name=bad
         user_team_name=_"Evil"
         [ai]
@@ -53,9 +54,9 @@
         {FLAG_VARIANT6 ragged}
     [/side]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Direwolf Rider" 3}
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Direwolf Rider" {ON_DIFFICULTY 1 2 3}}
 
-    {STARTING_VILLAGES 2 8}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 4 6 8}}
 
     [side]
         type=Troll Warrior
@@ -66,7 +67,7 @@
         controller=ai
         recruit=Troll,Troll Warrior,Goblin Pillager
         {GOLD 100 175 225}
-        {INCOME 20 25 30}
+        {INCOME 10 20 30}
         team_name=bad
         user_team_name=_"Evil"
         [ai]
@@ -74,9 +75,9 @@
         [/ai]
     [/side]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 4 "Troll Warrior" 3}
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 3 "Troll Warrior" {ON_DIFFICULTY 1 2 3}}
 
-    {STARTING_VILLAGES 3 8}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 4 6 8}}
 
     [side]
         type=Orcish Warlord
@@ -86,9 +87,13 @@
         canrecruit=yes
         facing=sw
         controller=ai
+#ifdef EASY
+        recruit=Orcish Crossbowman,Orcish Assassin,Orcish Warrior
+#else
         recruit=Orcish Crossbowman,Orcish Slayer,Orcish Warrior
+#endif
         {GOLD 100 150 200}
-        {INCOME 20 25 30}
+        {INCOME 10 20 30}
         team_name=bad
         user_team_name=_"Evil"
         [ai]
@@ -97,7 +102,7 @@
         {FLAG_VARIANT6 ragged}
     [/side]
 
-    {STARTING_VILLAGES 4 8}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 4 6 8}}
 
     [event]
         name=prestart
@@ -121,11 +126,7 @@
 #define TROLL_GUARD SIDE X Y WML
     [unit]
         side={SIDE}
-#ifdef EASY
-        type=Troll
-#else
-        type=Troll Warrior
-#endif
+        type={ON_DIFFICULTY "Troll Whelp" "Troll" "Troll Warrior"}
         generate_name=yes
         x,y={X},{Y}
         ai_special=guardian
@@ -183,12 +184,23 @@
             id=Engineer
             name= _ "Engineer"
             unrenamable=yes
-            type=Red Mage
+            type={ON_DIFFICULTY "Arch Mage" "Silver Mage" "Red Mage"}
             x,y=18,5
             side=1
             random_traits=no
             [modifications]
                 {TRAIT_LOYAL}
+#ifdef EASY
+                [object]
+                    silent=yes
+                    duration=forever
+                    [effect]
+                        apply_to=hitpoints
+                        increase_total=10
+                        heal_full=yes
+                    [/effect]
+                [/object]
+#endif
             [/modifications]
             {IS_LOYAL}
         [/unit]
