--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Sceptre_of_Fire/scenarios/6_Towards_the_Caves.cfg	2020-09-06 20:56:29.000000000 -0400
+++ campaigns/Sceptre_of_Fire/6_Towards_the_Caves.cfg	2020-08-29 22:43:38.000000000 -0400
@@ -2,7 +2,7 @@
 [scenario]
     name= _ "Towards the Caves"
     id=6_Towards_the_Caves
-    turns=12
+    {TURNS 20 16 12}
     map_data="{campaigns/Sceptre_of_Fire/maps/6_Towards_the_Caves.map}"
     next_scenario=7_Outriding_the_Outriders
 
@@ -17,7 +17,7 @@
         canrecruit=yes
         controller=human
         recruit=Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Guardsman,Dwarvish Scout,Gryphon Rider
-        gold=200
+        {GOLD 300 250 200}
         team_name=dwarves
         user_team_name= _ "Dwarves"
         {FLAG_VARIANT knalgan}
@@ -34,7 +34,7 @@
         side=2
         canrecruit=yes
         recruit=Elvish Rider,Elvish Hero,Elvish Marksman,Elvish Ranger,Elvish Druid
-        {GOLD 300 350 400}
+        {GOLD 250 350 400}
         team_name=elves
         user_team_name= _ "Elves"
         [ai]
@@ -71,7 +71,7 @@
         {FLAG_VARIANT wood-elvish}
     [/side]
 
-    {STARTING_VILLAGES 2 12}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 8 10 12}}
 
     [side]
         type=Elvish Marshal
@@ -81,7 +81,7 @@
         canrecruit=yes
         recruit=Elvish Rider,Elvish Hero,Elvish Marksman,Elvish Ranger,Elvish Druid
 
-        {GOLD 225 275 325}
+        {GOLD 200 275 325}
         team_name=elves
         user_team_name= _ "Elves"
         [ai]
@@ -90,15 +90,15 @@
                 [criteria]
                     x,y=36,3
                 [/criteria]
-                value=100
-                protect_radius=10
+                {QUANTITY value 60 80 100}
+                {QUANTITY protect_radius 8 9 10}
             [/goal]
             [goal]
                 name=target
                 [criteria]
                     race=dwarf
                 [/criteria]
-                value=10
+                {QUANTITY value 8 9 10}
             [/goal]
             [avoid]
                 x=1-18
@@ -108,8 +108,8 @@
         {FLAG_VARIANT wood-elvish}
     [/side]
 
-    {STARTING_VILLAGES 3 8}
-    {STARTING_VILLAGES_AREA 3 34 10 8}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 6 7 8}}
+    {STARTING_VILLAGES_AREA 3 34 10 {ON_DIFFICULTY 6 7 8}}
 
     [side]
         type=Elvish Captain
@@ -118,7 +118,7 @@
         side=4
         canrecruit=yes
         recruit=Elvish Scout,Elvish Fighter,Elvish Archer,Elvish Shaman
-        {GOLD 100 150 200}
+        {GOLD 75 150 200}
         team_name=elves
         user_team_name= _ "Elves"
         [ai]
@@ -127,7 +127,7 @@
                 [criteria]
                     id=Alanin
                 [/criteria]
-                value=10
+                {QUANTITY value 6 8 10}
             [/goal]
             [goal]
                 name=target
@@ -140,7 +140,7 @@
         {FLAG_VARIANT wood-elvish}
     [/side]
 
-    {STARTING_VILLAGES 4 11}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 9 10 11}}
 
     [story]
         [part]
@@ -471,6 +471,50 @@
             speaker=Alanin
             message= _ "I suppose..."
         [/message]
+#ifdef EASY
+        [message]
+            speaker=Thursagan
+            message= _ "Here, if you are still having doubts, let me inscribe this rune for you to help you slip past their defenses."
+        [/message]
+        [modify_unit]
+            [filter]
+                id=Alanin
+            [/filter]
+            [object]
+                silent=yes
+                duration=forever
+                [effect]
+                    apply_to=movement
+                    increase=1
+                [/effect]
+                [effect]
+                    apply_to=new_ability
+                    [abilities]
+                        {ABILITY_CONCEALMENT}
+                        {ABILITY_SKIRMISHER}
+                    [/abilities]
+                [/effect]
+            [/object]
+        [/modify_unit]
+        [sound]
+            name=mace.wav
+        [/sound]
+        [heal_unit]
+            [filter]
+                id=Alanin
+            [/filter]
+            moves=full
+            restore_attacks=yes
+        [/heal_unit]
+        [message]
+            speaker=Alanin
+            message= _ "Thanks!"
+        [/message]
+        [message]
+            speaker=narrator
+            message= _ "Alanin has gained new abilities!"
+        [/message]
+#endif
         [message]
             speaker=Gaenlar
             message= _ "Come out, dwarves, and surrender! Or die in that cave, your choice."
@@ -481,15 +525,15 @@
         name=turn 4
         [gold]
             side=2
-            amount=200
+            {QUANTITY amount 80 140 200}
         [/gold]
         [gold]
             side=3
-            amount=200
+            {QUANTITY amount 80 140 200}
         [/gold]
         [gold]
             side=4
-            amount=200
+            {QUANTITY amount 60 130 200}
         [/gold]
     [/event]
 
