--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Secrets_of_the_Ancients/scenarios/12_Walking_Trees.cfg	2021-09-05 22:32:23.000000000 -0400
+++ 12_Walking_Trees.cfg	2021-05-09 00:44:47.000000000 -0400
@@ -18,6 +18,7 @@
                 image=story/elf-road.jpg
                 scale=no
             [/background_layer]
+            # po: the "elf sorceress" is actually of the type Elvish Enchantress, not Elvish Sorceress
             story= _ "17 IV, 23 YW
 
 My primary objective was met, and I had been able to press a number of ghosts into service. Also, I had a new companion with a talent for healing. I doubted that Ras-Tabahn was already waiting at the appointed place, but I still wanted to hurry, so we followed the elf road north as quickly as we could. Fortunately, it was not in bad repair, though in many places it was overgrown, leaving only a few smooth cobbles to mark the way. The second day, we were still deep in the woods when we encountered an elf sorceress."
@@ -27,12 +28,12 @@
 
     {DEFAULT_SCHEDULE}
     {DEFAULT_MUSIC_PLAYLIST}
-    {TURNS 18 18 18}
+    {TURNS 20 19 18}
     victory_when_enemies_defeated=no
 
     [side]
         {SIDE_1_ARDONNA}
-        {GOLD 200 190 180}
+        {GOLD 218 199 180}
     [/side]
 
     [side]
@@ -60,19 +61,19 @@
         {FLAG_VARIANT wood-elvish}
     [/side]
 
-    # For each difficulty level, the first 16 woses are level-1, the rest are level-2.
+    # For each difficulty level, the first {18 17 16} woses are level-1, the rest are level-2.
 #ifdef EASY
 #define MAX_NUMBER_OF_WOSES
-    18
+    17 # yes, this is less than the number L2 woses start spawning at; I think it's ok to disallow them completely on EASY
 #enddef
 #define WOSES_PER_TURN
-    4
+    3
 #enddef
 #endif
 
 #ifdef NORMAL
 #define MAX_NUMBER_OF_WOSES
-    22
+    21
 #enddef
 #define WOSES_PER_TURN
     4
@@ -161,7 +162,7 @@
                             [and]
                                 x=$wose_x
                                 y=$wose_y
-                                radius=7  # Too close to the player’s keep.
+                                radius={ON_DIFFICULTY 9 8 7}  # No good. Too close to the player’s keep.
                             [/and]
                         [/or]
 
@@ -172,9 +173,11 @@
                             [and]
                                 x=$wose_x
                                 y=$wose_y
-                                radius=2
-                                # The wose would be adjacent to one of the
-                                # player's units after it moved one hex.
+                                radius={ON_DIFFICULTY 4 3 2}
+                                # No good. The wose would be adjacent to
+                                # one of the player's units after it moved
+                                # {3 2 1} hex(es).
+                                # FIXME: do I have this correct?
                             [/and]
                         [/or]
 
@@ -221,7 +224,7 @@
                 [if]
                     [variable]
                         name=number_of_woses
-                        greater_than=16
+                        greater_than={ON_DIFFICULTY 18 17 16} # prevents Elder Woses from being selected on EASY
                     [/variable]
                     [then]
                         [set_variable]
