--- /Users/ericgallager/Downloads/To_Lands_Unknown/scenarios/19_Sky_Kingdom.cfg	2019-05-19 15:00:26.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/To_Lands_Unknown/19_Sky_Kingdom.cfg	2020-09-01 23:58:36.000000000 -0400
@@ -4,7 +4,7 @@
     next_scenario=20_Enlightenment
     name= _ "Sky Kingdom"
     map_data="{~add-ons/To_Lands_Unknown/maps/19_Sky_Kingdom.map}"
-    {TURNS 40 35 32}
+    {TURNS 45 35 32}
     victory_when_enemies_defeated=no
 
     {SCENARIO_MUSIC breaking_the_chains.ogg}
@@ -48,7 +48,7 @@
 
         recruit=EoMa_Mage_of_Air,EoMa_Subversive_Mage,EoMa_Elementalist,EoMa_Golem,EoMa_Mu,EoMa_Hidden_Face,EoMa_Mage_of_Water,EoMa_Black_Mage,EoMa_Mystic_Warrior,EoMa_Golem_Boss
 
-        {GOLD 125 175 250}
+        {GOLD 120 175 250}
         {INCOME -1 0 1}
         village_gold=0
 
@@ -71,7 +71,7 @@
 
         recruit=EoMa_Mage_of_Air,EoMa_Battlemage,EoMa_Subversive_Mage,EoMa_Golem,EoMa_Mu,EoMa_Hidden_Face,EoMa_War_Mage,EoMa_Sorcerer,EoMa_Mystic_Warrior,EoMa_Cosmic_Eye
 
-        {GOLD 150 225 300}
+        {GOLD 140 225 300}
         {INCOME -1 -0 1}
         village_gold=0
 
