--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Under_the_Burning_Suns/scenarios/11_Battle_for_Zocthanol_Isle.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Under_the_Burning_Suns/scenarios/11_Battle_for_Zocthanol_Isle.cfg	2020-09-10 22:49:07.000000000 -0400
@@ -15,7 +15,7 @@
     {EXTRA_SCENARIO_MUSIC "heroes_rite.ogg"}
 
     victory_when_enemies_defeated=no
-    {TURNS 44 40 36}
+    {TURNS 48 42 36}
 
     {TWO_SUNS_DEFAULT_SCHEDULE}
 
@@ -61,8 +61,8 @@
         id=Kaleh
         type=Quenoth Youth
         canrecruit=yes
-        gold=200
-        {NO_INCOME}
+        {GOLD 220 210 200}
+        {INCOME 3 2 1}
         controller=human
         shroud=yes
         fog=yes
@@ -82,7 +82,7 @@
             {TRAIT_INTELLIGENT}
             {TRAIT_RESILIENT}
         [/modifications]
-        # gold and income will be set later - this ensures saurians have no gold until then
+        # gold and income will be set later - this ensures saurians have no gold until then:
         gold=0
         income=-50
         controller=ai
@@ -122,8 +122,8 @@
         id=Kelur
         name= _ "Kelur"
         canrecruit=yes
-        {GOLD 125 150 150}
-        {INCOME 15 17 19}
+        {GOLD 100 125 150}
+        {INCOME 10 14 18}
         controller=ai
         team_name=eloh_ally
         user_team_name=_"Eloh Cultists"
@@ -143,8 +143,8 @@
 
         [ai]
             recruitment_pattern=scout,fighter,archer,fighter
-            aggression=0.8
-            caution=0.1
+            aggression={ON_DIFFICULTY 0.6 0.7 0.8}
+            caution={ON_DIFFICULTY 0.3 0.2 0.1}
         [/ai]
     [/side]
 
@@ -159,8 +159,8 @@
             {TRAIT_STRONG}
             {TRAIT_RESILIENT}
         [/modifications]
-        {GOLD 125 150 150}
-        {INCOME 15 17 19}
+        {GOLD 100 125 150}
+        {INCOME 10 14 18}
         controller=ai
         team_name=eloh_ally
         user_team_name=_"Eloh Cultists"
@@ -180,8 +180,8 @@
 
         [ai]
             recruitment_pattern=scout, fighter, archer, fighter, mixed fighter
-            aggression=0.8
-            caution=0.1
+            aggression={ON_DIFFICULTY 0.6 0.7 0.8}
+            caution={ON_DIFFICULTY 0.3 0.2 0.1}
         [/ai]
     [/side]
 
@@ -210,7 +210,9 @@
     [side]
         side=7
         no_leader=yes
+        # gold and income will be set later - this ensures naga have no gold until then:
         gold=0
+        {NO_INCOME}
         controller=ai
         team_name=eloh_ally
         user_team_name=_"Eloh Cultists"
@@ -229,8 +231,8 @@
 #endif
 
         [ai]
-            aggression=0.75
-            caution=0.1
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            caution={ON_DIFFICULTY 0.3 0.2 0.1}
 
             recruitment_pattern=fighter, archer, fighter, fighter
 
@@ -244,7 +246,9 @@
     [side]
         side=8
         no_leader=yes
+        # gold and income will be set later - this ensures naga have no gold until then:
         gold=0
+        {NO_INCOME}
         controller=ai
         team_name=eloh_ally
         user_team_name=_"Eloh Cultists"
@@ -263,8 +267,8 @@
 #endif
 
         [ai]
-            aggression=0.75
-            caution=0.1
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            caution={ON_DIFFICULTY 0.3 0.2 0.1}
 
             recruitment_pattern=fighter, archer, fighter, fighter
 
@@ -684,8 +688,8 @@
 
         [modify_side]
             side=7
-            {INCOME 7 9 11}
-            {GOLD 75 95 115}
+            {INCOME 6 8 10}
+            {GOLD 70 90 110}
         [/modify_side]
 
         [unit]
@@ -702,8 +706,8 @@
 
         [modify_side]
             side=8
-            {INCOME 9 11 13}
-            {GOLD 75 95 115}
+            {INCOME 8 10 12}
+            {GOLD 70 90 110}
         [/modify_side]
     [/event]
 
@@ -824,6 +828,20 @@
             side=1
         [/allow_recruit]
 
+        [message]
+            race=merman
+            message= _ "Also, here is some gold."
+        [/message]
+
+        [gold]
+            side=1
+            amount={ON_DIFFICULTY 30 20 10}
+        [/gold]
+
+        [sound]
+            name=gold.ogg
+        [/sound]
+
         [remove_event]
             id=jungle_warning
         [/remove_event]
@@ -1192,7 +1210,7 @@
             name= _ "Grateful Merman"
             x,y=10,5
             side=1
-            hitpoints=12
+            hitpoints={ON_DIFFICULTY 18 15 12} # out of a max of 60
             unrenamable=yes
             [modifications]
                 {TRAIT_LOYAL}
@@ -1240,7 +1258,7 @@
         [/message]
 
         [gold]
-            amount=150
+            amount={ON_DIFFICULTY 170 160 150}
             side=1
         [/gold]
     [/event]
@@ -1482,14 +1500,14 @@
 
         # create vampire bats and blood bats
 # wmlindent: start ignoring
-        {SCATTER_UNITS {ON_DIFFICULTY 5 3 3} (Vampire Bat) 2 (terrain=Gs^Ftr) (
+        {SCATTER_UNITS {ON_DIFFICULTY 4 3 2} (Vampire Bat) 2 (terrain=Gs^Ftr) (
             side=6
             name= _ "Nocturnal Pest"
             upkeep=free)}
 
         # create blood bats on medium and hard difficulties
 #ifndef EASY
-        {SCATTER_UNITS {ON_DIFFICULTY 0 3 4} (Blood Bat) 2 (terrain=Gs^Ftr) (
+        {SCATTER_UNITS {ON_DIFFICULTY 0 3 5} (Blood Bat) 2 (terrain=Gs^Ftr) (
             side=6
             name= _ "Nocturnal Pest"
             upkeep=free)}
