--- /Users/ericgallager/Documents/GitHub/Genesis/episode1/scenarios/05_Crypt.cfg	2021-11-30 22:23:29.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/Genesis/05_Crypt.cfg	2020-11-13 01:30:01.000000000 -0500
@@ -2325,7 +2325,7 @@
         [filter_condition]
             [variable]
                 name=turn_counter
-                numerical_equals=7
+                numerical_equals={ON_DIFFICULTY 9 8 7}
             [/variable]
         [/filter_condition]
 
@@ -2337,6 +2337,8 @@
             id=s5l4_2
             random_gender=yes
             x,y=21,33
+            placement=map
+            passable=yes
         [/unit]
         [unit]
             side=5
@@ -2346,6 +2348,8 @@
             id=s5l5_2
             random_gender=yes
             x,y=20,38
+            placement=map
+            passable=yes
         [/unit]
         [unit]
             side=6
@@ -2355,6 +2359,8 @@
             id=s5l6_2
             random_gender=yes
             x,y=43,40
+            placement=map
+            passable=yes
         [/unit]
         [unit]
             side=7
@@ -2364,6 +2370,8 @@
             id=s5l7_2
             random_gender=yes
             x,y=43,34
+            placement=map
+            passable=yes
         [/unit]
         [unit]
             side=8
@@ -2373,6 +2381,8 @@
             id=s5l8_2
             random_gender=yes
             x,y=36,25
+            placement=map
+            passable=yes
         [/unit]
 
         [message]
