--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/The_Rise_Of_Wesnoth/scenarios/17c_Troll_Hole.cfg	2021-02-09 13:29:33.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/The_Rise_Of_Wesnoth/scenarios/17c_Troll_Hole.cfg	2021-03-07 22:01:15.000000000 -0500
@@ -5,7 +5,7 @@
     next_scenario=16_The_Kalian
     map_data="{campaigns/The_Rise_Of_Wesnoth/maps/17c_Troll_Hole.map}"
 
-    {TURNS 39 36 33}
+    {TURNS 45 42 39}
     {UNDERGROUND}
 
     {SCENARIO_MUSIC underground.ogg}
@@ -22,12 +22,12 @@
         unrenamable=yes
         side=1
         canrecruit=yes
-        gold=200
+        {GOLD 280 240 200}
         controller=human
         shroud=yes
         team_name=Haldric
         user_team_name=_"Refugees"
-        {INCOME 8 4 2}
+        {INCOME 8 5 2}
         {FLAG_VARIANT loyalist}
     [/side]
 
@@ -38,19 +38,19 @@
         side=2
         canrecruit=yes
         {GOLD 80 140 200}
-        {INCOME 2 4 8}
+        {INCOME 2 5 8}
         team_name=Troll
         user_team_name=_"Trolls"
         recruit=Troll Whelp,Troll,Troll Rocklobber
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=fighter,fighter,mixed fighter
-            aggression=0.75
-            {ATTACK_DEPTH 3 5 5}
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            {ATTACK_DEPTH 3 4 5}
         [/ai]
     [/side]
 
-    {STARTING_VILLAGES 2 10}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Troll Warrior
@@ -59,19 +59,19 @@
         side=3
         canrecruit=yes
         {GOLD 80 140 200}
-        {INCOME 2 4 8}
+        {INCOME 2 5 8}
         team_name=Troll
         user_team_name=_"Trolls"
         recruit=Troll Whelp,Troll,Troll Rocklobber
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=fighter,fighter,mixed fighter
-            aggression=0.75
-            {ATTACK_DEPTH 3 5 5}
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            {ATTACK_DEPTH 3 4 5}
         [/ai]
     [/side]
 
-    {STARTING_VILLAGES 3 10}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Troll Warrior
@@ -80,19 +80,19 @@
         side=4
         canrecruit=yes
         {GOLD 80 140 200}
-        {INCOME 2 4 8}
+        {INCOME 2 5 8}
         team_name=Troll
         user_team_name=_"Trolls"
         recruit=Troll Whelp,Troll,Troll Rocklobber
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=fighter,fighter,mixed fighter
-            aggression=0.75
-            {ATTACK_DEPTH 3 5 5}
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            {ATTACK_DEPTH 3 4 5}
         [/ai]
     [/side]
 
-    {STARTING_VILLAGES 4 10}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Troll Warrior
@@ -101,27 +101,22 @@
         side=5
         canrecruit=yes
         {GOLD 80 140 200}
-        {INCOME 2 4 8}
+        {INCOME 2 5 8}
         team_name=Troll
         user_team_name=_"Trolls"
         recruit=Troll Whelp,Troll,Troll Rocklobber
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=fighter,fighter,mixed fighter
-            aggression=0.75
-            {ATTACK_DEPTH 3 5 5}
+            aggression={ON_DIFFICULTY 0.69 0.72 0.75}
+            {ATTACK_DEPTH 3 4 5}
         [/ai]
     [/side]
 
-    {STARTING_VILLAGES 5 10}
+    {STARTING_VILLAGES 5 {ON_DIFFICULTY 8 9 10}}
 
-#ifdef EASY
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll" 1}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll Rocklobber" 1}
-#else
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll" 2}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll Rocklobber" 2}
-#endif
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll" {ON_DIFFICULTY 1 2 3}}
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 2,3,4,5 "Troll Rocklobber" {ON_DIFFICULTY 1 2 3}}
 
     {campaigns/The_Rise_Of_Wesnoth/utils/trow-nlmsg.cfg}
 
@@ -140,14 +135,31 @@
         {LOYAL_UNIT 4 (Tentacle of the Deep) 11 15}
         {LOYAL_UNIT 5 (Tentacle of the Deep) 19 17}
 #endif
-#ifdef HARD
 
+#ifdef HARD
         {LOYAL_UNIT 4 (Tentacle of the Deep) 11 15}
         {LOYAL_UNIT 5 (Tentacle of the Deep) 19 17}
         {LOYAL_UNIT 2 (Tentacle of the Deep) 11 17}
         {LOYAL_UNIT 3 (Tentacle of the Deep) 19 15}
 #endif
 
+        # FIXME: this doesn't appear to work?
+#define SPIDER_TRAIT
+        [+unit]
+            [modifications]
+#ifdef EASY
+                {TRAIT_AGED}
+#endif
+#ifdef NORMAL
+                {TRAIT_SLOW}
+#endif
+#ifdef HARD
+                {TRAIT_RESILIENT}
+#endif
+            [/modifications]
+        [/unit]
+#enddef
+
         {RANDOM 1..4}
         [if]
             [variable]
@@ -158,11 +170,11 @@
                 [kill]
                     id=Erart
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 3 5 5}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 3 5 5} {SPIDER_TRAIT}
                 [kill]
                     id=Raol
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 2 25 5}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 2 25 5} {SPIDER_TRAIT}
             [/then]
         [/if]
         [if]
@@ -174,11 +186,11 @@
                 [kill]
                     id=Raol
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 2 25 5}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 2 25 5} {SPIDER_TRAIT}
                 [kill]
                     id=Rilg
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 5 5 25}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 5 5 25} {SPIDER_TRAIT}
             [/then]
         [/if]
         [if]
@@ -190,11 +202,11 @@
                 [kill]
                     id=Rilg
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 5 5 25}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 5 5 25} {SPIDER_TRAIT}
                 [kill]
                     id=Gulg
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 4 25 25}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 4 25 25} {SPIDER_TRAIT}
             [/then]
         [/if]
         [if]
@@ -206,11 +218,11 @@
                 [kill]
                     id=Gulg
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 4 25 25}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 4 25 25} {SPIDER_TRAIT}
                 [kill]
                     id=Erart
                 [/kill]
-                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 3 5 5}
+                {GUARD_UNIT (Giant Spider) () ( _ "Fang") 3 5 5} {SPIDER_TRAIT}
             [/then]
         [/if]
 
@@ -249,7 +261,7 @@
 
             [gold_carryover]
                 bonus=yes
-                carryover_percentage=40
+                carryover_percentage={ON_DIFFICULTY 60 50 40}
             [/gold_carryover]
         [/objectives]
     [/event]
@@ -298,7 +310,7 @@
         [filter]
             id=Erart
         [/filter]
-        {LOOT 50 1}
+        {LOOT {ON_DIFFICULTY 70 60 50} 1}
     [/event]
 
     [event]
@@ -306,7 +318,7 @@
         [filter]
             id=Raol
         [/filter]
-        {LOOT 50 1}
+        {LOOT {ON_DIFFICULTY 70 60 50} 1}
     [/event]
 
     [event]
@@ -314,7 +326,7 @@
         [filter]
             id=Rilg
         [/filter]
-        {LOOT 50 1}
+        {LOOT {ON_DIFFICULTY 70 60 50} 1}
     [/event]
 
     [event]
@@ -322,7 +334,7 @@
         [filter]
             id=Gulg
         [/filter]
-        {LOOT 50 1}
+        {LOOT {ON_DIFFICULTY 70 60 50} 1}
     [/event]
 
     [event]
@@ -349,7 +361,7 @@
         [endlevel]
             result=victory
             bonus=yes
-            {NEW_GOLD_CARRYOVER 40}
+            {NEW_GOLD_CARRYOVER {ON_DIFFICULTY 60 50 40}}
         [/endlevel]
     [/event]
 [/scenario]
