--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/The_Rise_Of_Wesnoth/scenarios/20_Return_of_the_Fleet.cfg	2021-02-09 13:29:33.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/The_Rise_Of_Wesnoth/scenarios/20_Return_of_the_Fleet.cfg	2021-03-11 21:36:21.000000000 -0500
@@ -5,7 +5,7 @@
     next_scenario=21_The_Plan
     map_data="{campaigns/The_Rise_Of_Wesnoth/maps/20_Return_of_the_Fleet.map}"
 
-    {TURNS 45 42 39}
+    {TURNS 49 44 39}
     {DEFAULT_SCHEDULE}
 
     {SCENARIO_MUSIC vengeful.ogg}
@@ -25,14 +25,15 @@
         type=Noble Commander
         unrenamable=yes
         canrecruit=yes
-        gold=200
+        {GOLD 220 210 200}
+        {INCOME 3 2 1}
         controller=human
         team_name=Haldric
         user_team_name=_"Refugees"
         {FLAG_VARIANT loyalist}
     [/side]
 
-    {STARTING_VILLAGES 1 5}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 7 6 5}}
 
     [side]
         type=Draug
@@ -62,7 +63,7 @@
             passive_leader=yes
             recruitment_pattern=scout,fighter,archer
             aggression=0.4
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -71,7 +72,7 @@
         {INCOME 2 4 8}
     [/side]
 
-    {STARTING_VILLAGES 2 10}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -94,9 +95,8 @@
         team_name=orcs
         user_team_name=_"Orcs"
         [ai]
-            #{NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,mixed fighter,archer
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -107,7 +107,7 @@
         {INCOME 2 4 8}
     [/side]
 
-    {STARTING_VILLAGES 3 10}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -131,9 +131,8 @@
         team_name=orcs
         user_team_name=_"Orcs"
         [ai]
-            #{NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,mixed fighter,archer
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -144,7 +143,7 @@
         {INCOME 2 4 8}
     [/side]
 
-    {STARTING_VILLAGES 4 10}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 8 9 10}}
 
     [event]
         name=prestart
@@ -463,7 +462,7 @@
             [/filter_location]
         [/filter]
 
-        #No campaign is complete without a cuttlefish
+        # No campaign is complete without a cuttlefish:
         {LOYAL_UNIT 3 (Cuttle Fish) 11 17}
 #ifdef NORMAL
         {LOYAL_UNIT 3 (Cuttle Fish) 11 18}
@@ -570,7 +569,7 @@
 
         [gold]
             side=4
-            amount=80
+            amount={ON_DIFFICULTY 60 70 80}
         [/gold]
     [/event]
 
@@ -591,7 +590,7 @@
 
         [gold]
             side=3
-            amount=80
+            amount={ON_DIFFICULTY 60 70 80}
         [/gold]
     [/event]
 [/scenario]
