--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/The_Rise_Of_Wesnoth/scenarios/22_The_Rise_of_Wesnoth.cfg	2021-02-09 13:29:33.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/The_Rise_Of_Wesnoth/scenarios/22_The_Rise_of_Wesnoth.cfg	2021-03-11 22:12:45.000000000 -0500
@@ -6,7 +6,7 @@
     victory_when_enemies_defeated=no
     map_data="{campaigns/The_Rise_Of_Wesnoth/maps/22_The_Rise_of_Wesnoth.map}"
 
-    {TURNS 45 42 39}
+    {TURNS 49 44 39}
     {DEFAULT_SCHEDULE}
 
     {SCENARIO_MUSIC knalgan_theme.ogg}
@@ -26,13 +26,16 @@
         type=Noble Commander
         unrenamable=yes
         canrecruit=yes
-        gold=200
+        {GOLD 440 320 200}
+        {INCOME 9 6 3}
         controller=human
         team_name=Haldric
         user_team_name=_"Refugees"
         {FLAG_VARIANT loyalist}
     [/side]
 
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 8 6 4}}
+
     #2 is Jevyan
     #3 is the Sea Castle
     #4 is the Coast Orc
@@ -47,14 +50,14 @@
         side=2
         canrecruit=yes
         recruit=Revenant,Deathblade,Bone Shooter,Skeleton,Skeleton Archer,Vampire Bat,Walking Corpse,Soulless,Vampire Bat,Ghoul,Necrophage
-        {GOLD 160 180 210}
+        {GOLD 160 180 200}
         user_team_name=_"Evil"
         team_name=orcs
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,archer,scout,fighter,fighter,archer
             passive_leader=yes
-            {ATTACK_DEPTH 5 6 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -74,7 +77,7 @@
         {FLAG_VARIANT undead}
     [/side]
 
-    {STARTING_VILLAGES 2 10}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -83,13 +86,13 @@
         side=3
         canrecruit=yes
         recruit=Naga Fighter,Vampire Bat
-        {GOLD 130 150 180}
+        {GOLD 130 150 170}
         team_name=orcs
         user_team_name=_"Evil"
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=fighter,fighter,scout
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -97,10 +100,10 @@
             caution=0.0
             grouping=no
         [/ai]
-        {INCOME 1 2 4}
+        {INCOME 1 2 3}
     [/side]
 
-    {STARTING_VILLAGES 3 10}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -120,24 +123,23 @@
 #ifdef HARD
         recruit=Orcish Archer, Orcish Assassin, Orcish Grunt, Wolf Rider, Orcish Crossbowman, Orcish Warrior, Goblin Knight, Goblin Pillager, Orcish Slayer, Troll Whelp, Troll, Troll Warrior, Troll Rocklobber, Goblin Spearman, Goblin Impaler
 #endif
-        {GOLD 170 190 220}
+        {GOLD 170 190 210}
         team_name=orcs
         user_team_name=_"Evil"
         [ai]
-            #{NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,archer,scout,fighter,fighter,archer,mixed fighter
             grouping=defensive
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
             aggression=0.65
             grouping=offensive
         [/ai]
-        {INCOME 2 4 8}
+        {INCOME 2 4 6}
     [/side]
 
-    {STARTING_VILLAGES 4 10}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -157,14 +159,13 @@
 #ifdef HARD
         recruit=Orcish Archer, Orcish Assassin, Orcish Grunt, Wolf Rider, Orcish Crossbowman, Orcish Warrior, Goblin Knight, Goblin Pillager, Orcish Slayer, Troll Whelp, Troll, Troll Warrior, Troll Rocklobber, Goblin Spearman, Goblin Impaler
 #endif
-        {GOLD 150 170 200}
+        {GOLD 140 170 200}
         team_name=orcs
         user_team_name=_"Evil"
         [ai]
-            #{NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,archer,scout,fighter,fighter,archer,mixed fighter
             grouping=defensive
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -172,10 +173,10 @@
             grouping=offensive
         [/ai]
 
-        {INCOME 1 2 4}
+        {INCOME 1 2 3}
     [/side]
 
-    {STARTING_VILLAGES 5 10}
+    {STARTING_VILLAGES 5 {ON_DIFFICULTY 8 9 10}}
 
     [side]
         type=Orcish Warlord
@@ -195,13 +196,13 @@
 #ifdef HARD
         recruit=Orcish Archer, Orcish Assassin, Orcish Grunt, Wolf Rider, Orcish Crossbowman, Orcish Warrior, Goblin Knight, Goblin Pillager, Orcish Slayer, Troll Whelp, Troll, Troll Warrior, Troll Rocklobber, Goblin Spearman, Goblin Impaler
 #endif
-        {GOLD 150 170 200}
+        {GOLD 140 170 200}
         team_name=orcs
         user_team_name=_"Evil"
         [ai]
             {NO_SCOUTS}
             recruitment_pattern=scout,fighter,fighter,archer,scout,fighter,fighter,archer,mixed fighter
-            {ATTACK_DEPTH 5 5 6}
+            {ATTACK_DEPTH 4 5 6}
         [/ai]
         [ai]
             time_of_day=dusk,first_watch,second_watch
@@ -209,10 +210,10 @@
             caution=0.0
             grouping=no
         [/ai]
-        {INCOME 1 2 4}
+        {INCOME 1 2 3}
     [/side]
 
-    {STARTING_VILLAGES 6 10}
+    {STARTING_VILLAGES 6 {ON_DIFFICULTY 8 9 10}}
 
     [event]
         name=prestart
@@ -1041,7 +1042,11 @@
                 [terrain]
                     x=23,23
                     y=22,23
+#ifdef EASY
+                    terrain=Wwf^Es
+#else
                     terrain=Ww^Es
+#endif
                 [/terrain]
 
                 [redraw]
@@ -1162,7 +1167,11 @@
                 [terrain]
                     x=43,43
                     y=17,18
+#ifdef EASY
+                    terrain=Wwf^Es
+#else
                     terrain=Ww^Es
+#endif
                 [/terrain]
 
                 [redraw]
@@ -1274,7 +1283,11 @@
                 [terrain]
                     x=42,41
                     y=30,31
+#ifdef EASY
+                    terrain=Wwf^Es
+#else
                     terrain=Ww^Es
+#endif
                 [/terrain]
 
                 [redraw]
@@ -1324,7 +1337,6 @@
         {LOYAL_UNIT 2 (Naga Fighter) 2 35}
 #endif
 #ifdef HARD
-
         {LOYAL_UNIT 3 (Naga Fighter) 2 33}
         {LOYAL_UNIT 2 (Naga Fighter) 2 35}
         {LOYAL_UNIT 3 (Naga Fighter) 1 34}
@@ -1555,7 +1567,9 @@
         first_time_only=no
 
         {VARIABLE_OP next_corpse_spawn_turn rand "$($turn_number + 1)..$($turn_number + 5)"}
-        #{DEBUG_MSG "setting next corpse spawn to trigger on turn $next_corpse_spawn_turn"}
+#ifdef DEBUG_MODE
+        {DEBUG_MSG "setting next corpse spawn to trigger on turn $next_corpse_spawn_turn"}
+#endif
     [/event]
 
     # This triggers the corpse spawn on the turn determined by the above event.
@@ -1589,7 +1603,9 @@
         first_time_only=no
 
         {VARIABLE_OP next_skeleton_spawn_turn rand "$($turn_number + 1)..$($turn_number + 4)"}
-        #{DEBUG_MSG "setting next skeleton spawn to trigger on turn $next_skeleton_spawn_turn"}
+#ifdef DEBUG_MODE
+        {DEBUG_MSG "setting next skeleton spawn to trigger on turn $next_skeleton_spawn_turn"}
+#endif
     [/event]
 
     # This triggers the skeleton spawn on the turn determined by the above event.
