--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Dead_Water/scenarios/10_The_Flaming_Sword.cfg	2020-09-06 20:56:29.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Dead_Water/10_The_Flaming_Sword.cfg	2020-08-06 05:13:51.000000000 -0400
@@ -27,14 +27,15 @@
     {DW_TRACK {JOURNEY_10_NEW} }
 
     {DEFAULT_SCHEDULE}
-    {TURNS4 30 28 28 28}
+    {TURNS4 34 32 30 28}
     victory_when_enemies_defeated=no
 
     # wmllint: validate-off
     [side]
         {SIDE_1}
         fog=yes
-        {GOLD4 120 120 120 120}
+        {INCOME4 4 3 2 1}
+        {GOLD4 150 140 130 120}
     [/side]
     # wmllint: validate-on
 
@@ -66,8 +67,8 @@
         [/ai]
 
         {FLAG_VARIANT undead}
-        {INCOME4 30 35 40 45}
-        {GOLD4 500 600 700 800}
+        {INCOME4 25 30 35 40}
+        {GOLD4 400 500 600 700}
     [/side]
 
     {STARTING_VILLAGES 2 5}
@@ -140,8 +141,12 @@
 
 #ifdef EASY
         {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Wraith) 4}
-        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Shadow) 4}
+        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Shadow) 1}
         {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Spectre) 1}
+        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Deathblade) 1}
+        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Bone Shooter) 1}
+        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Revenant) 1}
+        {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Necrophage) 1}
 #endif
 #ifdef NORMAL
         {LIMIT_CONTEMPORANEOUS_RECRUITS 2 (Dread Bat) 2}
