--- /Applications/MacPorts/The Battle for Wesnoth.app/Contents/Resources/data/campaigns/Delfadors_Memoirs/scenarios/06_Swamps_of_Illuven.cfg	2020-09-06 20:56:30.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Delfadors_Memoirs/06_Swamps_of_Illuven.cfg	2020-12-21 00:59:19.000000000 -0500
@@ -4,7 +4,7 @@
     id=06_Swamps_of_Illuven
     next_scenario=07_Night_in_the_Swamp
     {MEMOIRS_MAP 06_The_Swamps_of_Illuven.map}
-    turns=28
+    {TURNS 32 30 28}
     {DEFAULT_SCHEDULE}
 
     {SCENARIO_MUSIC underground.ogg}    # No story part, so no intro music
@@ -24,6 +24,8 @@
         team_name=allies
         user_team_name= _ "Loyalists"
         {FLAG_VARIANT loyalist}
+        # 100 is the default when left unspecified:
+        {GOLD 120 110 100}
     [/side]
     [event]
         name=prestart
@@ -33,7 +35,7 @@
         [/recall]
         {MODIFY_UNIT (side=1) facing ne}
     [/event]
-    {STARTING_VILLAGES 1 4}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 6 5 4}}
 
     [side]
         type=Rogue
@@ -45,13 +47,13 @@
         recruit=Thief,Poacher,Thug
         team_name=enemies
         user_team_name= _ "Bandits"
-        {GOLD 80 100 150}
+        {GOLD 50 100 150}
         [ai]
             villages_per_scout=15
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
-    {STARTING_VILLAGES 2 4}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 2 3 4}}
 
     [side]
         type=Bandit
@@ -63,13 +65,13 @@
         recruit=Footpad,Poacher,Thug
         team_name=enemies
         user_team_name= _ "Bandits"
-        {GOLD 80 100 150}
+        {GOLD 50 100 150}
         [ai]
             villages_per_scout=10
         [/ai]
         {FLAG_VARIANT6 ragged}
     [/side]
-    {STARTING_VILLAGES 3 4}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 2 3 4}}
 
     victory_when_enemies_defeated=no
     {MEMOIRS_VICTORY_POSITION 38 3 yes}
@@ -128,20 +130,60 @@
             side=1
             variable=fee
         [/store_gold]
-        {VARIABLE_OP fee multiply 8}
+        {VARIABLE_OP fee multiply {ON_DIFFICULTY 6 7 8}}
         {VARIABLE_OP fee divide 10}
         {VARIABLE_OP fee round 0}
+        {VARIABLE garrath_hostile yes}
         [message]
             speaker=Garrath
             message=_"Greetings, strangers! This swamp is dangerous... You wanna cross it, you’ll need protection — cost you only $fee gold!"
             [option]
                 label=_"Thanks very much. Here’s the gold..."
                 [command]
-                    {VARIABLE_OP fee multiply -1}
                     [gold]
                         side=1
+                        amount=-$fee
+                    [/gold]
+                    # take his gold:
+                    [remove_item]
+                        x=4
+                        y=3
+                    [/remove_item]
+#ifdef EASY
+                    [message]
+                        speaker=Garrath
+                        message=_"You know, you really ought to be more careful before handing over money to suspicious strangers like me. But, since I'm feeling nice at the moment, I think I will actually keep my word and offer you some protection."
+                    [/message]
+                    [message]
+                        speaker=Harold
+                        message=_"Hey, what's your problem, Garrath? I thought we had a deal!"
+                    [/message]
+                    [message]
+                        speaker=Garrath
+                        message=_"Yeah, well, I changed my mind."
+                    [/message]
+                    [modify_side]
+                        side=2
+                        team_name=allies
+                    [/modify_side]
+                    {VARIABLE garrath_hostile no}
+                    [gold]
+                        side=2
                         amount=$fee
                     [/gold]
+                    # we are already inside an "ifdef EASY" block, so no need to vary this on difficulty:
+                    {CAPTURE_VILLAGES 2 4 3 4}
+                    [redraw]
+                    [/redraw]
+                    [message]
+                        speaker=Delfador
+                        message=_"...I'm not entirely sure what just happened here, but I guess I should be grateful?"
+                    [/message]
+                    [message]
+                        speaker=Lionel
+                        message=_"While it appears we lucked out this time, I fear for the travelers for whom this man does NOT keep his word..."
+                    [/message]
+#else
                     [message]
                         speaker=Garrath
                         message=_"It’s been a pleasure doing business with you... Bye!"
@@ -149,11 +191,6 @@
                     [kill]
                         side=2
                     [/kill]
-                    # take his gold too
-                    [remove_item]
-                        x=4
-                        y=3
-                    [/remove_item]
                     [redraw]
                     [/redraw]
                     [delay]
@@ -171,11 +208,222 @@
                         speaker=Lionel
                         message=_"Well, at least we only have to deal with one of them now..."
                     [/message]
+#endif
                     # give the remaining enemy a boost, for balance
                     [gold]
                         side=3
-                        amount=40
+                        amount={ON_DIFFICULTY 32 36 40}
                     [/gold]
+                    # (maybe also give them an income, but I can't decide on a good amount)
+#ifdef EASY
+                    [event]
+                        name=capture
+                        first_time_only=no
+                        [filter]
+                            side=1
+                        [/filter]
+                        [if]
+                            [have_location]
+                                x,y=$x1,$y1
+                                owner_side=2
+                            [/have_location]
+                            [or]
+                                [variable]
+                                    name=owner_side
+                                    equals=2
+                                [/variable]
+                            [/or]
+                            [then]
+                                [if]
+                                    [variable]
+                                        name=garrath_hostile
+                                        boolean_equals=no
+                                    [/variable]
+                                    [then]
+                                        # TODO: different dialogue if you've already given him an extra payment once
+                                        [message]
+                                            speaker=Garrath
+                                            message=_"Hey, that was one of OUR villages! Deal's off; we're no longer protecting you!"
+                                        [/message]
+                                        # Need to conditionalize here to ensure Garrath's reply makes sense:
+                                        [if]
+                                            [have_unit]
+                                                id=Harold
+                                            [/have_unit]
+                                            [then]
+                                                [message]
+                                                    speaker=Harold
+                                                    message=_"So... does that mean that OUR deal is back on now, Garrath?"
+                                                [/message]
+                                                [message]
+                                                    speaker=Garrath
+                                                    message=_"Fine, I guess..."
+                                                [/message]
+                                            [/then]
+                                        [/if]
+                                        [message]
+                                            speaker=Delfador
+                                            message=_"Wait, I'm sorry! I didn't realize you were so attached to that village! Please give me a chance to make it up to you!"
+                                        [/message]
+                                        [message]
+                                            speaker=Garrath
+                                            message=_"Well, I guess a little extra payment could persuade me to forgive you..."
+                                        [/message]
+                                        [store_gold]
+                                            variable=gold
+                                            side=1
+                                        [/store_gold]
+                                        [if]
+                                            [variable]
+                                                name=gold
+                                                less_than=20
+                                            [/variable]
+                                            [then]
+#ifdef DEBUG_MODE
+                                                [message]
+                                                    speaker=Delfador
+                                                    message=_"It looks like I only have $gold pieces of gold left..."
+                                                [/message]
+#else
+                                                [message]
+                                                    speaker=Delfador
+                                                    message=_"I'm afraid I don't have much gold left..."
+                                                [/message]
+#endif
+                                                [message]
+                                                    speaker=Garrath
+                                                    message=_"Oh well, too bad for you!"
+                                                [/message]
+                                                [modify_side]
+                                                    side=2
+                                                    team_name=enemies
+                                                [/modify_side]
+                                                {VARIABLE garrath_hostile yes}
+                                            [/then]
+                                            [else]
+                                                [message]
+                                                    speaker=Delfador
+                                                    message=_"Well..."
+                                                    [option]
+                                                        label=_"Here's 20 more gold."
+                                                        [command]
+                                                            [gold]
+                                                                side=1
+                                                                amount=-20
+                                                            [/gold]
+                                                            [gold]
+                                                                side=2
+                                                                amount=20
+                                                            [/gold]
+                                                            [message]
+                                                                speaker=Garrath
+                                                                message=_"Pleasure doing business with you!"
+                                                            [/message]
+                                                            # If Harold is dead, dialogue will still make sense with this omitted:
+                                                            [message]
+                                                                speaker=Harold
+                                                                message=_"Sheesh, Garrath, won't you make up your mind already?!"
+                                                            [/message]
+                                                            # (nothing else to do; he stays an ally)
+                                                        [/command]
+                                                    [/option]
+                                                    [option]
+                                                        label=_"Nope, I'm not giving you any more gold."
+                                                        [command]
+                                                            [message]
+                                                                speaker=Garrath
+                                                                message=_"Suit yourself. Men, attack!"
+                                                            [/message]
+                                                            [modify_side]
+                                                                side=2
+                                                                team_name=enemies
+                                                            [/modify_side]
+                                                            {VARIABLE garrath_hostile yes}
+                                                        [/command]
+                                                    [/option]
+                                                [/message]
+                                            [/else]
+                                        [/if] # gold
+                                        {CLEAR_VARIABLE gold}
+                                    [/then]
+                                    [else]
+                                        [message]
+                                            speaker=Garrath
+                                            message=_"Stop stealing my villages!"
+                                        [/message]
+                                    [/else]
+                                [/if] # garrath_hostile
+                            [/then]
+                        [/if]
+                    [/event]
+
+                    [event]
+                        name=side 1 turn refresh
+                        first_time_only=no
+                        [store_gold]
+                            variable=gold
+                            side=1
+                        [/store_gold]
+                        [if]
+                            [variable]
+                                name=gold
+                                less_than=0
+                            [/variable]
+                            [then]
+                                [message]
+                                    speaker=Garrath
+                                    message=_"Hm, it looks like you don't have any gold left. We're not interested in you anymore; bye!"
+                                [/message]
+                                [kill]
+                                    side=2
+                                [/kill]
+                                [message]
+                                    speaker=Harold
+                                    message=_"What a fickle one, that Garrath..."
+                                [/message]
+                            [/then]
+                        [/if]
+                        {CLEAR_VARIABLE gold}
+                    [/event]
+
+                    # A little bit of extra balance to make up for the fact that Harold now has an extra enemy:
+                    [event]
+                        name=last breath
+                        [filter]
+                            side=2
+                        [/filter]
+                        [filter_second]
+                            side=3
+                        [/filter_second]
+                        [message]
+                            speaker=Harold
+                            message=_"I'll be taking back that money that your men ""borrowed"" from me now, Garrath!"
+                        [/message]
+                    [/event]
+
+                    # Split the actual gold-giving part of this off from the previous one so that it can happen multiple times:
+                    [event]
+                        name=die
+                        first_time_only=no
+                        [filter]
+                            side=2
+                        [/filter]
+                        [filter_second]
+                            side=3
+                        [/filter_second]
+                        {RANDOM $unit.level..$unit.cost}
+                        [gold]
+                            side=3
+                            amount=$random
+                        [/gold]
+                        [floating_text]
+                            x,y=$x2,$y2
+                            # wmllint: markcheck off
+                            text="<span color='#BCB088'>+$random|g</span>"
+                            # wmllint: markcheck on
+                        [/floating_text]
+                    [/event]
+#endif
                 [/command]
             [/option]
             [option]
@@ -208,7 +456,7 @@
                         [/sound]
                         [gold]
                             side=1
-                            amount=100
+                            amount={ON_DIFFICULTY 120 110 100}
                         [/gold]
                         [message]
                             speaker=unit
@@ -219,5 +467,6 @@
             [/option]
         [/message]
         {CLEAR_VARIABLE fee}
+        {CLEAR_VARIABLE garrath_hostile}
     [/event]
 [/scenario]
