--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode3/scenarios/03_Amidst_the_Ruins_of_Glamdrol.cfg	2021-02-15 18:34:30.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/03_Amidst_the_Ruins_of_Glamdrol.cfg	2021-11-24 01:00:15.000000000 -0500
@@ -4,7 +4,7 @@
     id=03_Amidst_the_Ruins_of_Glamdrol
     name= _ "Amidst the Ruins of Glamdrol"
     {MAP 03_Amidst_the_Ruins_of_Glamdrol.map}
-    {TURNS 26 24 22}
+    {TURNS 28 25 22}
     next_scenario=04_01_Outpost_of_Hell
 
     {G_DEATHS}
@@ -26,7 +26,8 @@
         team_name=good
         user_team_name= _ "team_name^Elynia"
 
-        {GOLD 155 145 140}
+        {GOLD 195 165 140}
+        {INCOME 3 2 1}
 
         # wmllint: recognize Elynia
         {CHARACTER_STATS_ELYNIA}
@@ -41,7 +42,8 @@
         user_team_name= _ "team_name^General Bardyl"
         {ALLIANCE_FLAG}
 
-        {GOLD 210 200 190}
+        {GOLD 290 240 190}
+        {INCOME 3 2 1}
         recruit=Aragwaith Spearman,Aragwaith Archer,Aragwaith Scout,Orcish Grunt,Orcish Archer,Wolf Rider,Orcish Assassin
 
         # wmllint: recognize General Bardyl
@@ -49,14 +51,14 @@
 
         [unit]
             x,y=18,13
-            experience=97
+            experience={ON_DIFFICULTY 99 98 97}
             # wmllint: recognize Kyara
             {CHARACTER_STATS_KYARA}
         [/unit]
 
         [unit]
             x,y=19,13
-            experience=142
+            experience={ON_DIFFICULTY 144 143 142}
             # wmllint: recognize Horo
             {CHARACTER_STATS_HORO}
         [/unit]
@@ -71,6 +73,9 @@
             [modifications]
                 {TRAIT_LOYAL}
                 {TRAIT_RESILIENT}
+#ifdef EASY
+                {TRAIT_HEALTHY}
+#endif
             [/modifications]
         [/unit]
 
@@ -84,6 +89,9 @@
             [modifications]
                 {TRAIT_LOYAL}
                 {TRAIT_INTELLIGENT}
+#ifdef EASY
+                {TRAIT_STRONG}
+#endif
             [/modifications]
         [/unit]
 
@@ -131,7 +139,7 @@
         user_team_name= _ "team_name^Nar-hamoth"
         {CHAOS_FLAG}
 
-        {GOLD 180 190 200}
+        {GOLD 160 180 200}
         recruit=Demon,Demon Zephyr,Shaxthal Rayblade,Shaxthal Protector Drone,Chaos Invader,Chaos Headhunter,Chaos Invoker
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -197,7 +205,7 @@
             name= _ "Cervan"
             x,y=6,42
             upkeep=loyal
-            #ai_special=guardian
+            ai_special=guardian
             facing=ne
             [modifications]
                 {TRAIT_INTELLIGENT}
@@ -210,7 +218,7 @@
             type=Chaos Marksman
             x,y=16,35
             upkeep=loyal
-            #ai_special=guardian
+            ai_special=guardian
             facing=ne
             [modifications]
                 {TRAIT_RESILIENT}
@@ -222,7 +230,7 @@
             type=Chaos Marksman
             x,y=14,34
             upkeep=loyal
-            #ai_special=guardian
+            ai_special=guardian
             facing=ne
             [modifications]
                 {TRAIT_STRONG}
@@ -302,7 +310,7 @@
         user_team_name= _ "team_name^Enemies"
         {CHAOS_FLAG}
 
-        {GOLD 175 180 185}
+        {GOLD 155 170 185}
         recruit=Chaos Bowman,Chaos Raider,Chaos Invader,Shaxthal Runner Drone
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -439,13 +447,13 @@
 #endif
     [/side]
 
-    {STARTING_VILLAGES 1 6}
-    {STARTING_VILLAGES 2 6}
-    {STARTING_VILLAGES 3 8}
-    {STARTING_VILLAGES 4 6}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 8 7 6}}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 8 7 6}}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 6 7 8}}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 4 5 6}}
 
     {LIMIT_CONTEMPORANEOUS_RECRUITS 3 ("Demon Zephyr")                        1 }
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 3 ("Shaxthal Rayblade")        {DIFF 1 2 2} }
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 3 ("Shaxthal Rayblade")        {DIFF 1 2 3} }
     {LIMIT_CONTEMPORANEOUS_RECRUITS 3 ("Shaxthal Protector Drone")            1 }
 
     [event]
@@ -494,7 +502,7 @@
         [/set_variables]
 
         [repeat]
-            times={DIFF 3 3 2}
+            times={DIFF 4 3 2}
             [do]
                 [apply_amlas]
                     id="Kyara"
@@ -507,7 +515,7 @@
         [/repeat]
 
         [repeat]
-            times={DIFF 2 2 1}
+            times={DIFF 3 2 1}
             [do]
                 [apply_amlas]
                     id="Horo"
