--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode3/scenarios/04_01_Outpost_of_Hell.cfg	2021-02-15 18:34:30.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/04_01_Outpost_of_Hell.cfg	2021-11-24 17:30:40.000000000 -0500
@@ -59,7 +59,8 @@
         team_name=good
         user_team_name= _ "team_name^Elynia"
 
-        {GOLD 170 160 150}
+        {GOLD 190 170 150}
+        {INCOME 3 2 1}
 
         # wmllint: recognize Elynia
         {CHARACTER_STATS_ELYNIA}
@@ -74,7 +75,8 @@
         user_team_name= _ "team_name^General Bardyl"
         {ALLIANCE_FLAG}
 
-        {GOLD 170 160 150}
+        {GOLD 190 170 150}
+        {INCOME 3 2 1}
 
         # wmllint: recognize General Bardyl
         {CHARACTER_STATS_GENERAL_BARDYL}
@@ -182,7 +184,7 @@
         [/modifications]
     [/side]
 
-    {LIMIT_RECRUITS 6 (Shaxthal Rayblade) ({DIFF 1 2 2})}
+    {LIMIT_RECRUITS 6 (Shaxthal Rayblade) ({DIFF 1 2 3})}
 
     [side]
         side=7
@@ -210,7 +212,7 @@
         [/modifications]
     [/side]
 
-    {LIMIT_RECRUITS 7 (Ixthala Fire Dancer) ({DIFF 1 2 2})}
+    {LIMIT_RECRUITS 7 (Ixthala Fire Dancer) ({DIFF 1 2 3})}
 
     [side]
         side=8
@@ -218,7 +220,7 @@
         user_team_name= _ "team_name^Aran-Balgur"
         {CHAOS_FLAG}
 
-        {GOLD 180 190 200}
+        {GOLD 160 180 200}
         recruit=Chaos Hound,Valdemon Basher,Blood Imp,Chaos Bowman,Shaxthal Runner Drone
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -237,7 +239,7 @@
         [/modifications]
     [/side]
 
-    {LIMIT_RECRUITS 8 (Valdemon Basher) ({DIFF 1 2 2})}
+    {LIMIT_RECRUITS 8 (Valdemon Basher) ({DIFF 1 2 3})}
 
     [side]
         side=9
@@ -245,8 +247,8 @@
         user_team_name= _ "team_name^Aran-Balgur"
         {CHAOS_FLAG}
 
-        {GOLD 150 160 170}
-        {INCOME 6 8 10}
+        {GOLD 140 155 170}
+        {INCOME 4 7 10}
         recruit=Imp,Blood Imp,Valdemon Basher,Serpent Messenger,Demon Zephyr,Shaxthal Runner Drone,Shaxthal Protector Drone,Chaos Crossbowman,Chaos Bowman,Chaos Raider
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -356,7 +358,7 @@
     [/side]
 
     # NOTE: yes, this limits these recruits as a set, not individually.
-    {LIMIT_RECRUITS 9 (Valdemon Basher,Serpent Messenger,Demon Zephyr,Shaxthal Protector Drone) ({DIFF 3 3 4})}
+    {LIMIT_RECRUITS 9 (Valdemon Basher,Serpent Messenger,Demon Zephyr,Shaxthal Protector Drone) ({DIFF 2 3 4})}
 
     # Let side 9 recruit only every even turn.
 
@@ -2100,7 +2102,7 @@
         {VARIABLE_RANDOM shapeshifter_flag 1..9001}
         # Easier difficulties have smaller chances of getting
         # shapeshifters, probably.
-        {VARIABLE_MOD shapeshifter_flag ({DIFF 5 3 2})}
+        {VARIABLE_MOD shapeshifter_flag ({DIFF 6 4 2})}
 
         # Don't allow more than a certain amount of shapeshifters
         # to exist on map at the same time.
