--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode2/scenarios/05_The_Eastern_Front.cfg	2020-10-17 19:54:05.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/05_The_Eastern_Front.cfg	2021-11-20 12:49:03.000000000 -0500
@@ -23,7 +23,9 @@
         team_name=good
         user_team_name= _ "team_name^Elynia"
 
-        {GOLD 250 250 240}
+        {GOLD 298 269 240}
+        {INCOME 9 5 1}
+        village_gold={ON_DIFFICULTY 3 2 1}
 
         # wmllint: recognize Elynia
         {CHARACTER_STATS_ELYNIA}
@@ -31,6 +33,7 @@
     # wmllint: validate-on
 
     [side]
+        # Southeast
         side=2
         team_name=evil
         user_team_name= _ "team_name^Elves"
@@ -41,7 +44,8 @@
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
-            {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "scout,archer,healer,fighter,mixed fighter,scout,archer"}
+            {AI_SIMPLE_ALWAYS_ASPECT recruitment_more "Shaxthal Rayblade"}
+            {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "fighter,scout,archer,healer,fighter,mixed fighter,scout,archer"}
         [/ai]
 
         canrecruit=yes
@@ -53,16 +57,22 @@
             {TRAIT_DEXTROUS}
             {TRAIT_RESILIENT}
         [/modifications]
+#ifdef EASY
+        [status]
+            slowed=yes
+        [/status]
+#endif
     [/side]
 
     [side]
+        # Northeast
         side=3
         team_name=evil
         user_team_name= _ "team_name^Elves"
         {CHAOS_FLAG}
 
         recruit=Elvish Archer,Elvish Fighter,Elvish Scout,Elvish Shaman
-        {GOLD 170 180 190}
+        {GOLD 130 160 190}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -83,6 +93,7 @@
     [/side]
 
     [side]
+        # Northwest
         side=4
         team_name=evil
         user_team_name= _ "team_name^Elves"
@@ -102,6 +113,7 @@
         id=Unidë
         name= _ "Unidë"
         facing=se
+        moves={ON_DIFFICULTY 0 1 2} # (this will only apply on the first turn, but that's intentional)
         [modifications]
             {TRAIT_RESILIENT}
             {TRAIT_QUICK}
@@ -142,10 +154,10 @@
 
     {UNDEAD_RECRUITMENT:ZYNARA:WILD_GHOSTS_SIDE 6}
 
-    {STARTING_VILLAGES 1 5}
-    {STARTING_VILLAGES 2 7}
-    {STARTING_VILLAGES 3 5}
-    {STARTING_VILLAGES 4 8}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 7 6 5}}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 5 6 7}}
+    {STARTING_VILLAGES 3 {ON_DIFFICULTY 4 5 6}}
+    {STARTING_VILLAGES 4 {ON_DIFFICULTY 6 7 8}}
 
     [event]
         name=prestart
@@ -195,11 +207,38 @@
             message= _ "There are so many different flowers in this field! It’s beautiful!"
         [/message]
 
+        [modify_unit]
+            [filter]
+                id=Anya
+            [/filter]
+            [object]
+                id=Anyas_floral_fortification
+                duration=forever
+                silent=yes
+                [effect]
+                    apply_to=hitpoints
+                    increase={ON_DIFFICULTY 3 2 1}
+                    increase_total={ON_DIFFICULTY 3 2 1}
+                    heal_full=yes
+                [/effect]
+            [/object]
+        [/modify_unit]
+
         [message]
             speaker=Elynia
             message= _ "Indeed it is."
         [/message]
 
+        [message]
+            speaker=Anya
+            message= _ "The windmills, too, make the scene quite picturesque... it's a shame that these windmills don't seem to function as villages, like some others do..."
+        [/message]
+
+        [message]
+            speaker=Elynia
+            message= _ "The villages where the windmillkeepers live are close nearby to the windmills, so it's no great loss."
+        [/message]
+
         [move_unit_fake]
             side=1
             type=$durvan_store.type
@@ -277,8 +316,15 @@
             message= _ "May the Lords of Light protect us..."
         [/message]
 
+        [modify_unit]
+            [filter]
+                id=Unidë
+            [/filter]
+            moves={ON_DIFFICULTY 1 2 3} # (again, only applies 1 turn, which is intentional)
+        [/modify_unit]
+
         [scroll_to]
-            x,y=8,4
+            x,y=8,4 # Unidë's keep
         [/scroll_to]
 
         [message]
@@ -288,7 +334,7 @@
         [/message]
 
         [scroll_to]
-            x,y=42,35
+            x,y=42,35 # Telan's keep
         [/scroll_to]
 
         [message]
@@ -298,7 +344,7 @@
         [/message]
 
         [scroll_to]
-            x,y=44,12
+            x,y=44,12 # Ethanin-Adrael's keep
         [/scroll_to]
 
         [message]
@@ -320,6 +366,16 @@
             speaker=Elynia
             message= _ "I fear for the fate of the elves who sent that call for help to the Council..."
         [/message]
+
+        [event]
+            name=side 4 turn 2 refresh
+            [modify_unit]
+                [filter]
+                    id=Unidë
+                [/filter]
+                moves={ON_DIFFICULTY 1 2 3} # (again, only applies 1 turn, which is intentional)
+            [/modify_unit]
+        [/event]
     [/event]
 
     [event]
@@ -567,6 +623,38 @@
         [/message]
     [/event]
 
+    # Justifications for these units:
+    # 1. Elves control 3 of the 4 corners of the map, leading players to wonder what's in the 4th one. They deserve rewards for investigating.
+    # 2. Players who like to scout will be left at a disadvantage since their units will be busy scouting instead of fighting; these units compensate for that.
+    # 3. Elynia has dialogue above saying that they're "close to Aragwaithi territory"; these units show that.
+    [event]
+        name=moveto
+        [filter]
+            side=1
+            x,y=4,34
+        [/filter]
+        # TODO: dialogue
+        {GENERIC_UNIT 1 (Aragwaith Archer) $x1 $y1}
+    [/event]
+    [event]
+        name=moveto
+        [filter]
+            side=1
+            x,y=9,39
+        [/filter]
+        # TODO: dialogue
+        {GENERIC_UNIT 1 (Aragwaith Eagle Rider) $x1 $y1}
+    [/event]
+    [event]
+        name=moveto
+        [filter]
+            side=1
+            x,y=12,40
+        [/filter]
+        # TODO: dialogue
+        {GENERIC_UNIT 1 (Aragwaith Scout) $x1 $y1}
+    [/event]
+
     [event]
         name=enemies defeated
         {ENDLEVEL_VICTORY yes}
