--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode3/scenarios/07A_Dark_Fire_part_1.cfg	2021-11-30 05:51:57.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/07A_Dark_Fire_part_1.cfg	2021-11-25 00:39:37.000000000 -0500
@@ -501,7 +501,11 @@
 
     [event]
         name=setup floor 1
-        #first_time_only=no
+#ifdef __DEBUG__
+#ifdef DEBUG_MODE
+        first_time_only=no
+#endif
+#endif
 
         # Dummy access guards; not actually playing since their
         # side isn't AI-controlled.
@@ -518,8 +522,10 @@
 
         {NAMED_LOYAL_UNIT 4 "Demon Spelldancer"  22  2 Litharen ( _ "Litharen")} {FACING sw} {GUARDIAN}
 
-        #{FINALE_A_GUARDIAN 4 (Demon Warrior) 25  8} {FACING se}
-        #{FINALE_A_GUARDIAN 4 (Demon Zephyr)  22  2} {FACING sw}
+#ifdef NIGHTMARE
+        {FINALE_A_GUARDIAN 4 (Demon Warrior) 25  8} {FACING se}
+        {FINALE_A_GUARDIAN 4 (Demon Zephyr)  22  2} {FACING sw}
+#endif
 
         [store_starting_location]
             side=4
@@ -534,8 +540,13 @@
             name= _ "Romeldan"
             facing=s
             [modifications]
+#ifdef EASY
+                {TRAIT_DIM}
+                {TRAIT_AGED}
+#else
                 {TRAIT_STRONG}
                 {TRAIT_QUICK}
+#endif
             [/modifications]
         [/unit]
 
@@ -552,8 +563,13 @@
             name= _ "Captain Verathorn"
             facing=sw
             [modifications]
+#ifdef EASY
+                {TRAIT_WEAK}
+                {TRAIT_SLOW}
+#else
                 {TRAIT_RESILIENT}
                 {TRAIT_INTELLIGENT}
+#endif
             [/modifications]
         [/unit]
 
@@ -562,10 +578,19 @@
             x,y=20,10
             facing=sw
             type=Chaos Hound
+#ifdef EASY
+            upkeep=full
+#else
             upkeep=loyal
+#endif
             [modifications]
+#ifdef EASY
+                {TRAIT_WEAK}
+                {TRAIT_FERAL_MUSTHAVE}
+#else
                 {TRAIT_QUICK}
                 {TRAIT_RESILIENT}
+#endif
             [/modifications]
         [/unit]
 
@@ -580,7 +605,11 @@
             upkeep=loyal
             [modifications]
                 {TRAIT_SLOW}
+#ifdef EASY
+                {TRAIT_WEAK}
+#else
                 {TRAIT_RESILIENT}
+#endif
             [/modifications]
         [/unit]
 
@@ -594,8 +623,13 @@
             gender=male
             upkeep=loyal
             [modifications]
+#ifdef EASY
+                {TRAIT_SLOW}
+                {TRAIT_DIM}
+#else
                 {TRAIT_STRONG}
                 {TRAIT_DEXTROUS}
+#endif
             [/modifications]
         [/unit]
 
@@ -609,8 +643,13 @@
             gender=male
             upkeep=loyal
             [modifications]
+#ifdef EASY
+                {TRAIT_WEAK}
+                {TRAIT_DIM}
+#else
                 {TRAIT_RESILIENT}
                 {TRAIT_QUICK}
+#endif
             [/modifications]
         [/unit]
 
@@ -624,8 +663,13 @@
             gender=male
             upkeep=loyal
             [modifications]
+#ifdef EASY
+                {TRAIT_SLOW}
+                {TRAIT_AGED}
+#else
                 {TRAIT_STRONG}
                 {TRAIT_INTELLIGENT}
+#endif
             [/modifications]
         [/unit]
 
@@ -834,10 +878,20 @@
 #define FINALE_A_ACTIVATE_SIDE _SIDE
     [modify_side]
         side={_SIDE}
-        {GOLD 80 90 100}
-        {INCOME 2 3 4}
+        {GOLD 60 80 100}
+        {INCOME 1 3 5}
         hidden=no
     [/modify_side]
+#ifdef EASY
+    [modify_unit]
+        [filter]
+            side={_SIDE}
+        [/filter]
+        [status]
+            slowed=yes
+        [/status]
+    [/modify_unit]
+#endif
 #enddef
 
 #define FINALE_A_RECRUITMENT_MECHANICS_NOTE
@@ -1136,6 +1190,17 @@
 
         {FINALE_A_ACTIVATE_SIDE 4}
 
+#ifdef EASY
+        [modify_unit]
+            [filter]
+                side=6
+            [/filter]
+            [status]
+                slowed=yes
+            [/status]
+        [/modify_unit]
+#endif
+
         {DISABLE_UNLIMITED_PLAYER_MOVES}
 
         {ALLOW_END_TURN}
@@ -1159,14 +1224,14 @@
         {UNLOCK_VIEW}
 
         #
-        # Make player hits always land for the first two turns.
+        # Make player hits always land for the first {4 3 2} turns.
         #
 
         {VARIABLE finale_a_player_cth yes}
 
         [event]
             delayed_variable_substitution=no
-            name="turn $($turn_number + 2)"
+            name="turn $($turn_number + {ON_DIFFICULTY 4 3 2})"
 
             {VARIABLE finale_a_player_cth no}
         [/event]
