--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode2/scenarios/08_And_then_there_was_Chaos.cfg	2021-11-30 05:51:57.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/08_And_then_there_was_Chaos.cfg	2021-11-20 17:19:12.000000000 -0500
@@ -42,7 +42,9 @@
         team_name=good
         user_team_name= _ "team_name^Elynia"
 
-        {GOLD 200 190 180}
+        {GOLD 218 199 180}
+        {INCOME 9 6 3}
+        village_gold={ON_DIFFICULTY 3 2 1}
 
         # wmllint: recognize Elynia
         {CHARACTER_STATS_ELYNIA}
@@ -55,12 +57,12 @@
         {CHAOS_FLAG}
 
         recruit=Elvish Acolyte,Elvish Fighter,Elvish Archer,Elvish Scout
-        {GOLD 140 150 160}
+        {GOLD 100 130 160}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
-            {AI_SIMPLE_ALWAYS_ASPECT aggression          0.80}
-            {AI_SIMPLE_ALWAYS_ASPECT caution             0.15}
+            {AI_SIMPLE_ALWAYS_ASPECT aggression          {ON_DIFFICULTY 0.78 0.79 0.80}}
+            {AI_SIMPLE_ALWAYS_ASPECT caution             {ON_DIFFICULTY 0.17 0.16 0.15}}
             {AI_SIMPLE_ALWAYS_ASPECT passive_leader      yes}
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "archer,scout,fighter,fighter,mixed fighter,archer,mixed fighter,scout,fighter"}
         [/ai]
@@ -84,7 +86,7 @@
         {CHAOS_FLAG}
 
         recruit=Demon,Imp,Chaos Headhunter,Chaos Invoker
-        {GOLD 150 175 190}
+        {GOLD 130 160 190}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -110,12 +112,12 @@
         {CHAOS_FLAG}
 
         recruit=Shaxthal Runner Drone,Chaos Hound,Chaos Headhunter,Chaos Invoker
-        {GOLD 150 160 170}
+        {GOLD 110 140 170}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
-            {AI_SIMPLE_ALWAYS_ASPECT aggression          0.90}
-            {AI_SIMPLE_ALWAYS_ASPECT caution             0.10}
+            {AI_SIMPLE_ALWAYS_ASPECT aggression          {ON_DIFFICULTY 0.88 0.89 0.90}}
+            {AI_SIMPLE_ALWAYS_ASPECT caution             {ON_DIFFICULTY 0.12 0.11 0.10}}
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "fighter,scout,fighter,mixed fighter,archer,mixed fighter,mixed fighter"}
         [/ai]
 
@@ -214,10 +216,12 @@
         {CHAOS_FLAG}
 
         recruit=Demon,Chaos Invader,Chaos Bowman,Chaos Raider
-        {GOLD 160 170 180}
+        {GOLD 120 150 180}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
+            {AI_SIMPLE_ALWAYS_ASPECT recruitment_diversity {ON_DIFFICULTY 4.4 3.3 2.2}}
+            {AI_SIMPLE_ALWAYS_ASPECT recruitment_randomness {ON_DIFFICULTY 99 88 77}}
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "scout,scout,archer,mixed fighter,archer,fighter,scout,fighter,scout"}
         [/ai]
 
@@ -254,7 +258,9 @@
 
     {STARTING_VILLAGES_ALL 2}
 
-    {STARTING_VILLAGES 1 7}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 9 8 7}} # you
+    # No point in varying these next 3, because otherwise they'd just go to side 2,
+    # which is also an enemy side:
     {STARTING_VILLAGES 3 4}
     {STARTING_VILLAGES 4 4}
     {STARTING_VILLAGES 5 7}
