--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode1/scenarios/10_Tears.cfg	2020-10-17 19:54:05.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/10_Tears.cfg	2020-10-11 22:11:05.000000000 -0400
@@ -46,7 +46,7 @@
     [side]
         side=2
         recruit=Chaos Invoker,Chaos Invader,Chaos Headhunter,Chaos Hound
-        {GOLD 230 250 270}
+        {GOLD 210 240 270}
         team_name=evil
         user_team_name= _ "team_name^Enemies"
         {CHAOS_FLAG}
@@ -67,7 +67,7 @@
             [/goal]
 
             {AI_SIMPLE_ALWAYS_ASPECT aggression 1.00}
-            {AI_SIMPLE_ALWAYS_ASPECT caution    0.05}
+            {AI_SIMPLE_ALWAYS_ASPECT caution    {ON_DIFFICULTY 0.09 0.07 0.05}}
         [/ai]
 
         type=Chaos Lorekeeper
@@ -92,7 +92,7 @@
     [side]
         side=3
         recruit=Chaos Raider,Chaos Bowman,Chaos Invader
-        {GOLD 220 230 240}
+        {GOLD 200 220 240}
         team_name=evil
         user_team_name= _ "team_name^Enemies"
         {CHAOS_FLAG}
@@ -113,7 +113,7 @@
             [/goal]
 
             {AI_SIMPLE_ALWAYS_ASPECT aggression 1.00}
-            {AI_SIMPLE_ALWAYS_ASPECT caution    0.05}
+            {AI_SIMPLE_ALWAYS_ASPECT caution    {ON_DIFFICULTY 0.09 0.07 0.05}}
         [/ai]
 
         type=Chaos Razerman
@@ -184,7 +184,8 @@
 
         [modify_side]
             side=1
-            gold="$("+{DIFF 240 220 200}+" + $main_side1_stats.gold)"
+            gold="$("+{DIFF 280 240 200}+" + $main_side1_stats.gold)"
+            {INCOME 3 2 1}
         [/modify_side]
 
         {CLEAR_VARIABLE main_side1_stats,original_recruit_list_strlen}
@@ -227,7 +228,7 @@
     # so village ownership has an effect at all.
     #
 
-    {STARTING_VILLAGES 1 5}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 7 6 5}}
 
     [event]
         name=start
