--- /Users/ericgallager/Documents/GitHub/After_the_Storm/episode1/scenarios/11_Return_to_Wesmere_part_1.cfg	2020-10-17 19:54:05.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_the_Storm/11_Return_to_Wesmere_part_1.cfg	2021-10-16 00:35:35.000000000 -0400
@@ -4,7 +4,7 @@
     id=11_Return_to_Wesmere_part_1
     name= _ "Return to Wesmere — Part 1"
     {MAP 11_Return_to_Wesmere_part_1.map}
-    {TURNS 34 32 30}
+    {TURNS 36 33 30}
     next_scenario="11_Return_to_Wesmere_part_1_cutscene"
 
     {B_DEATHS}
@@ -57,8 +57,8 @@
         team_name=player
         user_team_name= _ "team_name^Elynia"
 
-        {GOLD 360 350 340}
-        {INCOME 5 4 3}
+        {GOLD 398 369 340}
+        {INCOME 9 6 3}
 
         # wmllint: recognize Elynia
         {CHARACTER_STATS_ELYNIA_AS_LEADER}
@@ -70,7 +70,7 @@
         [criteria]
             id=Elynia
         [/criteria]
-        value=10.0
+        value={ON_DIFFICULTY 6.0 8.0 10.0}
     [/goal]
     [goal]
         [criteria]
@@ -83,7 +83,7 @@
     [side]
         side=2
         recruit=Chaos Invoker,Chaos Invader,Chaos Raider,Chaos Cavalier
-        {GOLD 170 195 210}
+        {GOLD 150 170 210}
         {INCOME 0 2 4}
         team_name=evil
         user_team_name= _ "team_name^Enemies"
@@ -139,7 +139,7 @@
     [side]
         side=4
         recruit=Demon,Demon Zephyr,Chaos Invoker,Chaos Invader,Shaxthal Rayblade
-        {GOLD 240 240 260}
+        {GOLD 200 230 260}
         team_name=evil
         user_team_name= _ "team_name^Wesmere Defense"
         {CHAOS_FLAG}
@@ -186,7 +186,7 @@
         [ai]
             {AI_SIMPLE_ALWAYS_ASPECT grouping     no}
             {AI_SIMPLE_ALWAYS_ASPECT aggression 1.00}
-            {AI_SIMPLE_ALWAYS_ASPECT caution    0.01}
+            {AI_SIMPLE_ALWAYS_ASPECT caution    {ON_DIFFICULTY 0.09 0.05 0.01}}
 
             {RTW1_ENEMY_TARGETS}
 
@@ -196,8 +196,8 @@
                     x=21
                     y=3
                 [/criteria]
-                value=5.0
-                protect_radius=10
+                value={ON_DIFFICULTY 3.0 4.0 5.0}
+                protect_radius={ON_DIFFICULTY 8 9 10}
             [/goal]
         [/ai]
 
@@ -247,13 +247,15 @@
 
     {STARTING_VILLAGES_ALL 4}
 
-    {STARTING_VILLAGES 1 4}
+    {STARTING_VILLAGES 1 {ON_DIFFICULTY 6 5 4}} # you
+    # no point in varying these next 2 since otherwise they'd just go to side 4, which is also
+    # an enemy side:
     {STARTING_VILLAGES 2 4}
     {STARTING_VILLAGES 3 4}
 
     {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Chaos Cavalier"                 ({DIFF 2 3 4})}
     {LIMIT_CONTEMPORANEOUS_RECRUITS 3 "Chaos Crossbowman"              ({DIFF 2 3 4})}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 4 "Demon Zephyr,Shaxthal Rayblade" ({DIFF 3 4 4})}
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 4 "Demon Zephyr,Shaxthal Rayblade" ({DIFF 2 3 4})}
 
     [event]
         name=prestart
