--- 05a_crossfire.cfg	2020-10-10 13:08:48.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/05a_Crossfire.cfg	2020-08-23 18:09:10.000000000 -0400
@@ -4,7 +4,7 @@
     id=05a_Crossfire
     name= _ "Crossfire"
     {MAP 05a_Crossfire.map}
-    {TURNS 33 31 30}
+    {TURNS 36 32 30}
     next_scenario=06_The_Moon_Valley
     victory_when_enemies_defeated=no
 
@@ -41,7 +41,8 @@
 
         shroud=yes
 
-        {GOLD 210 200 190}
+        {GOLD 220 200 190}
+        {INCOME 3 2 1}
 
         # wmllint: recognize Galas
         {CHARACTER_STATS_GALAS}
@@ -70,19 +71,22 @@
                 [criteria]
                     side=1
                 [/criteria]
-                value=8.0
+                # 1 = you, so increase with difficulty:
+                {QUANTITY value 6.0 7.0 8.0}
             [/goal]
             [goal]
                 [criteria]
                     side=4
                 [/criteria]
-                value=6.0
+                # 4 = Outlaws, so decrease with difficulty:
+                {QUANTITY value 8.0 7.0 6.0}
             [/goal]
             [goal]
                 [criteria]
                     side=3
                 [/criteria]
-                value=10.0
+                # 3 = Chaos Empire, so decrease with difficulty:
+                {QUANTITY value 12.0 11.0 10.0}
             [/goal]
         [/ai]
 
@@ -123,26 +127,26 @@
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
             {AI_SIMPLE_ALWAYS_ASPECT caution       0.25}
-            {AI_SIMPLE_ALWAYS_ASPECT aggression    0.92}
+            {AI_SIMPLE_ALWAYS_ASPECT aggression    {ON_DIFFICULTY 0.89 0.91 0.92}}
             {AI_SIMPLE_ALWAYS_ASPECT leader_value  6.0}
 
             [goal]
                 [criteria]
                     id=Mal Keshar
                 [/criteria]
-                value=8.0
+                {QUANTITY value 6.0 7.0 8.0}
             [/goal]
             [goal]
                 [criteria]
                     id=Anlindë
                 [/criteria]
-                value=9.0
+                {QUANTITY value 7.0 8.0 9.0}
             [/goal]
             [goal]
                 [criteria]
                     id=Galas
                 [/criteria]
-                value=10.0
+                {QUANTITY value 8.0 9.0 10.0}
             [/goal]
         [/ai]
 
