--- 07_Goliath.cfg	2020-10-10 13:11:54.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/07_Goliath.cfg	2020-08-23 23:06:13.000000000 -0400
@@ -4,7 +4,7 @@
     id=07_Goliath
     name= _ "scenario_name^Goliath"
     {MAP 07_Goliath.map}
-    {TURNS 48 45 42}
+    {TURNS 69 49 42}
     next_scenario=08_Errand_of_Hope
     victory_when_enemies_defeated=no
 
@@ -117,9 +117,13 @@
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
-            {AI_SIMPLE_ALWAYS_ASPECT aggression          0.99}
-            {AI_SIMPLE_ALWAYS_ASPECT caution             0.01}
+            {AI_SIMPLE_ALWAYS_ASPECT aggression          {ON_DIFFICULTY 0.33 0.66 0.99}}
+            {AI_SIMPLE_ALWAYS_ASPECT caution             {ON_DIFFICULTY 0.09 0.05 0.01}}
+#ifdef EASY
+            {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern scout,fighter,scout,fighter}
+#else
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern scout,fighter,scout,archer}
+#endif
         [/ai]
 
         canrecruit=yes
