--- 09_The_Library.cfg	2020-10-10 13:08:48.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/09_The_Library.cfg	2020-08-26 22:26:00.000000000 -0400
@@ -4,7 +4,7 @@
     id=09_The_Library
     name= _ "The Library"
     {MAP 09_The_Library.map}
-    {TURNS 50 49 48}
+    {TURNS 69 49 48}
     next_scenario=10_The_Source_of_Light
     victory_when_enemies_defeated=no
 
@@ -28,8 +28,8 @@
 
         shroud=yes
 
-        {GOLD 180 170 160}
-        {INCOME 4 3 3}
+        {GOLD 190 175 160}
+        {INCOME 5 4 3}
 
         # wmllint: recognize Galas
         {CHARACTER_STATS_GALAS}
@@ -53,7 +53,7 @@
         user_team_name= _ "team_name^Undead"
         {RAGGED_FLAG}
 
-        {GOLD 130 140 150}
+        {GOLD 110 130 150}
         recruit=Skeleton,Skeleton Archer,Vampire Bat,Blood Bat
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -80,7 +80,7 @@
         user_team_name= _ "team_name^Chaos Empire"
         {CHAOS_FLAG}
 
-        {GOLD 160 170 180}
+        {GOLD 140 160 180}
         recruit=Chaos Invader,Chaos Headhunter
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -88,6 +88,7 @@
             {AI_SIMPLE_ALWAYS_ASPECT aggression    1.00}
             {AI_SIMPLE_ALWAYS_ASPECT caution       0.00}
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "fighter,fighter,fighter,scout,fighter,scout"}
+            simple_targeting=yes
         [/ai]
 
         canrecruit=yes
@@ -122,7 +123,7 @@
         user_team_name= _ "team_name^Undead"
         {RAGGED_FLAG}
 
-        {GOLD 150 160 170}
+        {GOLD 130 150 170}
         recruit=Skeleton,Skeleton Rider,Skeleton Archer
 
         {ai/aliases/stable_singleplayer.cfg}
@@ -130,6 +131,7 @@
             {AI_SIMPLE_ALWAYS_ASPECT aggression    1.00}
             {AI_SIMPLE_ALWAYS_ASPECT caution       0.00}
             {AI_SIMPLE_ALWAYS_ASPECT recruitment_pattern "fighter,scout,scout,fighter,archer"}
+            simple_targeting=yes
         [/ai]
 
         canrecruit=yes
@@ -146,7 +148,11 @@
 
     [side]
         side=5
+#ifdef EASY
+        team_name=neutral
+#else
         team_name=evil
+#endif
         user_team_name= _ "team_name^Cave Creatures"
         {RAGGED_FLAG}
 
@@ -157,6 +163,7 @@
         [ai]
             {AI_SIMPLE_ALWAYS_ASPECT aggression    1.00}
             {AI_SIMPLE_ALWAYS_ASPECT caution       0.00}
+            simple_targeting=yes
         [/ai]
 
         {TL_CAVE_DWELLER (Fire Guardian) 13 41} {FACING se}
@@ -230,6 +237,7 @@
         [ai]
             {AI_SIMPLE_ALWAYS_ASPECT aggression  100.00} # Encourage them to get hurt.
             {AI_SIMPLE_ALWAYS_ASPECT caution       0.00}
+            simple_targeting=yes
         [/ai]
 
         {TL_CAVE_DWELLER (Shaxthal Worm) 54 43} {VARIATION a} {TL_PALETTE_OVERRIDE}
@@ -434,6 +442,7 @@
             [filter]
                 id=Galas
             [/filter]
+            # User-facing message telling about this is done farther below:
             [effect]
                 apply_to=movement_costs
                 [movement_costs]
@@ -578,6 +587,7 @@
         [message]
             speaker=narrator
             image=wesnoth-icon.png
+            # This is done with the silent object above:
             message= _ "As Galas has become accustomed to moving on rough cave terrains, his movement cost on them has decreased by one!"
         [/message]
 #endif
