--- 10_The_Source_of_Light.cfg	2020-10-10 13:11:54.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/10_The_Source_of_Light.cfg	2020-10-10 13:47:01.000000000 -0400
@@ -4,7 +4,7 @@
     id=10_The_Source_of_Light
     name= _ "The Source of Light"
     {MAP 10_The_Source_of_Light.map}
-    {TURNS 60 59 58}
+    {TURNS 69 59 58}
     next_scenario=10x_Elynia
     victory_when_enemies_defeated=no
 
@@ -36,7 +36,7 @@
 
         shroud=yes
 
-        {GOLD 200 180 170}
+        {GOLD 210 190 170}
         {INCOME 5 4 3}
 
         # wmllint: recognize Galas
@@ -47,13 +47,19 @@
     # Chaos incursion remnants
     [side]
         side=2
+#ifdef EASY
+        team_name=evil
+#else
         team_name=gaia
+#endif
         user_team_name= _ "team_name^Chaos Empire"
         {CHAOS_FLAG}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
-            #{AI_SIMPLE_ALWAYS_ASPECT leader_ignores_keep yes}
+#ifdef MAKE_LEADER_IGNORE_KEEP
+            {AI_SIMPLE_ALWAYS_ASPECT leader_ignores_keep yes}
+#endif
             {AI_SIMPLE_ALWAYS_ASPECT leader_aggression   1.0}
             {AI_SIMPLE_ALWAYS_ASPECT aggression          1.0}
             {AI_SIMPLE_ALWAYS_ASPECT village_value       0.0}
