--- 17_Voice_of_the_Armageddon.cfg	2020-10-10 13:11:54.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/17_Voice_of_the_Armageddon.cfg	2020-08-27 21:50:36.000000000 -0400
@@ -4,7 +4,7 @@
     id=17_Voice_of_the_Armageddon
     name= _ "Voice of the Armageddon"
     {MAP 17_Voice_of_the_Armageddon.map}
-    {TURNS 25 24 23}
+    {TURNS 26 24 23}
     next_scenario=18_Ashes_of_Wesmere
 
     {SCENARIO_MUSIC       "climactic_contemplation.ogg"}
@@ -27,8 +27,8 @@
 
         shroud=yes
 
-        {GOLD 180 170 160}
-        {INCOME 5 5 4}
+        {GOLD 190 175 160}
+        {INCOME 6 5 4}
 
         # wmllint: recognize Galas
         {CHARACTER_STATS_GALAS}
@@ -101,7 +101,7 @@
         user_team_name= _ "team_name^Chaos Empire"
         {CHAOS_FLAG}
 
-        {GOLD 180 190 200}
+        {GOLD 170 190 200}
         recruit=Skeleton,Ghost,Skeleton Archer,Ghoul,Demon
 
         canrecruit=yes
@@ -143,8 +143,10 @@
             {AI_SIMPLE_ALWAYS_ASPECT leader_aggression      9.00}
             {AI_SIMPLE_ALWAYS_ASPECT caution                0.00}
             {AI_SIMPLE_ALWAYS_ASPECT village_value          0.00}
-            #{AI_SIMPLE_ALWAYS_ASPECT leader_ignores_keep     yes}
-            #{AI_SIMPLE_ALWAYS_ASPECT grouping                 no}
+#ifdef MAKE_AI_BE_STUPID
+            {AI_SIMPLE_ALWAYS_ASPECT leader_ignores_keep     yes}
+            {AI_SIMPLE_ALWAYS_ASPECT grouping                 no}
+#endif
 
             {AI_BOSS_TARGETING_ENGINE <<{"Galas","Elynia","Mal Keshar"}>>}
         [/ai]
@@ -157,10 +159,10 @@
         {SIDE_GENERIC_GUARD (Ghast) 18 32} {FACING ne}
         {SIDE_GENERIC_GUARD (Ghast) 23 35} {FACING ne}
 
-        {SIDE_GENERIC_GUARD (Elvish Forefather)             12 37} {FACING se}
-        {SIDE_GENERIC_GUARD ({DIFF Wraith Spectre Spectre}) 15 32} {FACING se}
-        {SIDE_GENERIC_GUARD (Elvish Forefather)             25 36} {FACING nw}
-        {SIDE_GENERIC_GUARD ({DIFF Wraith Spectre Spectre}) 23 39} {FACING nw}
+        {SIDE_GENERIC_GUARD ({DIFF (Elvish Warrior Spirit) (Elvish Ancestor) (Elvish Forefather)})             12 37} {FACING se}
+        {SIDE_GENERIC_GUARD ({DIFF Ghost Wraith Spectre}) 15 32} {FACING se}
+        {SIDE_GENERIC_GUARD ({DIFF (Elvish Warrior Spirit) (Elvish Ancestor) (Elvish Forefather)})             25 36} {FACING nw}
+        {SIDE_GENERIC_GUARD ({DIFF Ghost Wraith Spectre}) 23 39} {FACING nw}
     [/side]
     # wmllint: validate-on
 
