--- 19_Under_the_Sands.cfg	2020-10-10 13:08:48.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/19_Under_the_Sands.cfg	2020-08-27 22:48:44.000000000 -0400
@@ -4,7 +4,7 @@
     id=19_Under_the_Sands
     name= _ "Under the Sands"
     {MAP 19_Under_the_Sands.map}
-    {TURNS 66 65 64}
+    {TURNS 69 65 64}
     next_scenario=19x_Crossroads
     victory_when_enemies_defeated=no
 
@@ -32,7 +32,7 @@
 
         shroud=yes
 
-        {GOLD 160 150 140}
+        {GOLD 190 150 140}
 
         # wmllint: recognize Galas
         {CHARACTER_STATS_GALAS}
@@ -488,6 +488,38 @@
             speaker=Erathan
             message= _ "I sorely regret accepting the council’s mission to follow you people. This is a truly dreadful place to be for someone who is neither a necromancer nor a walking light source."
         [/message]
+#ifdef EASY
+        [message]
+            speaker=Elynia
+            message= _ "Here, let me help you with a spell that will make it easier for you to tough it out."
+        [/message]
+        [modify_unit]
+            [filter]
+                id=Erathan
+            [/filter]
+            [object]
+                silent=yes
+                duration=forever
+                [effect]
+                    apply_to=hitpoints
+                    increase_total=10
+                    heal_full=yes
+                [/effect]
+            [/object]
+            {TRAIT_HEALTHY}
+        [/modify_unit]
+        [heal_unit]
+            id=Erathan
+            amount=full
+            animate=yes
+            moves=full
+            restore_attacks=yes
+        [/heal_unit]
+        [message]
+            speaker=narrator
+            message= _ "Erathan's hitpoints have increased!"
+        [/message]
+#endif
     [/event]
 
     [event]
