--- 20_The_Heart.cfg	2020-10-10 13:08:48.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/20_The_Heart.cfg	2020-09-16 22:38:27.000000000 -0400
@@ -4,47 +4,47 @@
     id=20_The_Heart
     name= _ "The Heart"
     {MAP 20_The_Heart.map}
-    {TURNS 32 30 28}
+    {TURNS 35 32 28}
     next_scenario=21_Innuendo
 
     {SCENARIO_MUSIC       "battle-epic.ogg"}
     {EXTRA_SCENARIO_MUSIC "loyalists.ogg"}
 
-    {DUSK1}
-    {SHORTDARK}
-    {DAWN2}
-    {MORNING2}
-    {MIDDAY2}
-    {AFTERNOON2}
-    {DUSK2}
-    {LONGDARK1}
-    {LONGDARK1}
-    {LONGDARK1}
-    {LONGDARK1}
-    {LONGDARK1}
-    {LONGDARK1}
-    {LONGDARK2}
-    {LONGDARK2}
-    {LONGDARK2}
-    {LONGDARK2}
-    {LONGDARK2}
-    {LONGDARK2}
-    {LONGDARK3}
-    {LONGDARK3}
-    {LONGDARK3}
-    {LONGDARK3}
-    {LONGDARK3}
-    {LONGDARK3}
-    {LONGDARK4}
-    {LONGDARK4}
-    {LONGDARK4}
-    {LONGDARK4}
-    {LONGDARK4}
-    {LONGDARK4}
-    {DAWN1}
-    {MORNING1}
-    {MIDDAY1}
-    {AFTERNOON1}
+    {DUSK1} #1
+    {SHORTDARK} #2
+    {DAWN2} #3
+    {MORNING2} #4
+    {MIDDAY2} #5
+    {AFTERNOON2} #6
+    {DUSK2} #7
+    {LONGDARK1} #8
+    {LONGDARK1} #9
+    {LONGDARK1} #10
+    {LONGDARK1} #11
+    {LONGDARK1} #12
+    {LONGDARK1} #13
+    {LONGDARK2} #14
+    {LONGDARK2} #15
+    {LONGDARK2} #16
+    {LONGDARK2} #17
+    {LONGDARK2} #18
+    {LONGDARK2} #19
+    {LONGDARK3} #20
+    {LONGDARK3} #21
+    {LONGDARK3} #22
+    {LONGDARK3} #23
+    {LONGDARK3} #24
+    {LONGDARK3} #25
+    {LONGDARK4} #26
+    {LONGDARK4} #27
+    {LONGDARK4} #28
+    {LONGDARK4} #29
+    {LONGDARK4} #30
+    {LONGDARK4} #31
+    {DAWN1} #32
+    {MORNING1} #33
+    {MIDDAY1} #34
+    {AFTERNOON1} #35
 
     {SHAXTHAL_SET_SURFACE_VARIATIONS_FLAG}
 
@@ -62,8 +62,8 @@
         team_name=good
         user_team_name= _ "team_name^Galas"
 
-        {GOLD 320 310 300}
-        {INCOME 7 5 3}
+        {GOLD 344 322 300}
+        {INCOME 9 6 3}
 
         # wmllint: recognize Galas
         {CHARACTER_STATS_GALAS}
@@ -77,9 +77,9 @@
         user_team_name= _ "team_name^Dwarves"
         {DWARVISH_FLAG}
 
-        {GOLD 330 320 310}
-        {INCOME 8 6 4}
-        recruit=Dwarvish Runesmith,Dwarvish Flamethrower,Dwarvish Guardsman,Dwarvish Scout,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Gryphon Rider
+        {GOLD 350 330 310}
+        {INCOME 9 7 5}
+        recruit=Dwarvish Runesmith,Dwarvish Flamethrower,Dwarvish Guardsman,Dwarvish Scout,Dwarvish Fighter,Dwarvish Thunderer,Dwarvish Ulfserker,Gryphon Rider,Dwarvish Steamwheel
 
         # wmllint: recognize Althurin
         canrecruit=yes
@@ -90,21 +90,27 @@
 #define TH_ENEMY_AI_TARGETS
     [goal]
         [criteria]
+            id=Althurin
+        [/criteria]
+        value={ON_DIFFICULTY 5.0 6.0 7.0}
+    [/goal]
+    [goal]
+        [criteria]
             id=Galas
         [/criteria]
-        value=8.0
+        value={ON_DIFFICULTY 6.0 7.0 8.0}
     [/goal]
     [goal]
         [criteria]
             id=Elynia
         [/criteria]
-        value=9.0
+        value={ON_DIFFICULTY 7.0 8.0 9.0}
     [/goal]
     [goal]
         [criteria]
             id=Mal Keshar
         [/criteria]
-        value=10.0
+        value={ON_DIFFICULTY 8.0 9.0 10.0}
     [/goal]
 #enddef
 
@@ -115,8 +121,8 @@
         {CHAOS_FLAG}
 
         recruit=Demon,Blood Imp,Demon Grunt,Demon Zephyr,Chaos Invader,Chaos Invoker,Chaos Headhunter,Chaos Hound,Shaxthal Runner Drone,Shaxthal Protector Drone
-        {GOLD 320 340 360}
-        {INCOME 7 9 10}
+        {GOLD 300 330 360}
+        {INCOME 4 7 10}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -151,8 +157,8 @@
         {CHAOS_FLAG}
 
         recruit=Chaos Hound,Demonic Hound,Chaos Invoker,Chaos Magus,Chaos Headhunter,Abomination,Imp,Automaton,Iron Golem
-        {GOLD 320 340 375}
-        {INCOME 6 7 8}
+        {GOLD 300 320 375}
+        {INCOME 4 6 8}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -187,8 +193,8 @@
         {CHAOS_FLAG}
 
         recruit=Chaos Bowman,Chaos Crossbowman,Doom Guard,Chaos Headhunter,Chaos Raider,Chaos Cavalier,Demon Zephyr
-        {GOLD 320 340 360}
-        {INCOME 5 7 8}
+        {GOLD 300 330 360}
+        {INCOME 2 5 8}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -240,7 +246,7 @@
         # Has about 24 income initially from villages and is immune to the S19
         # enemy gold factor.
         recruit=Demon Zephyr,Shaxthal Rayblade,Demon Grunt,Chaos Invoker,Demon,Chaos Hound,Chaos Invader
-        {GOLD 180 200 220}
+        {GOLD 170 200 220}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -299,10 +305,10 @@
         {SIDE_GENERIC_GUARD (Demon Warrior) 37 30} {FACING se}
         {SIDE_GENERIC_GUARD (Demon Warrior) 39 27} {FACING se}
 
-        {SIDE_GENERIC_GUARD (Gutwrencher Imp) 39 32}
-        {SIDE_GENERIC_GUARD (Gutwrencher Imp) 46 29}
-        {SIDE_GENERIC_GUARD (Gutwrencher Imp) 47 33}
-        {SIDE_GENERIC_GUARD (Gutwrencher Imp) 37 22}
+        {SIDE_GENERIC_GUARD ({DIFF (Blood Imp) (Gutwrencher Imp) (Gutwrencher Imp)}) 39 32}
+        {SIDE_GENERIC_GUARD ({DIFF (Blood Imp) (Gutwrencher Imp) (Gutwrencher Imp)}) 46 29}
+        {SIDE_GENERIC_GUARD ({DIFF (Blood Imp) (Gutwrencher Imp) (Gutwrencher Imp)}) 47 33}
+        {SIDE_GENERIC_GUARD ({DIFF (Blood Imp) (Gutwrencher Imp) (Gutwrencher Imp)}) 37 22}
 
         {SIDE_GENERIC_UNIT (Demon Warrior) 45 27}
         {SIDE_GENERIC_UNIT (Demon Warrior) 31 23}
@@ -423,9 +429,11 @@
 
     {STARTING_VILLAGES_ALL 6}
 
-    {STARTING_VILLAGES 1 6}
-    {STARTING_VILLAGES 2 6}
+    {STARTING_VILLAGES 1 {DIFF 10 8 6}} #you
+    {STARTING_VILLAGES 2 {DIFF 10 8 6}} #dwarves
 
+    # Reducing enemy village capture radius with difficulty just gives them to a different
+    # enemy side (6) (see the STARTING_VILLAGES_ALL above), so just leave this as-is:
     {STARTING_VILLAGES 3 3}
     {STARTING_VILLAGES 4 5}
     {STARTING_VILLAGES 5 5}
@@ -513,8 +521,8 @@
             name=demon_spawning
             [literal]
                 next_turn=3
-                turns={DIFF 10 8 6}
-                population_limit_percent={DIFF 0.40 0.75 0.90}
+                turns={DIFF 12 9 6}
+                population_limit_percent={DIFF 0.30 0.70 0.90}
             [/literal]
         [/set_variables]
 
@@ -547,7 +555,11 @@
 
         [unit]
             side=2
+#ifdef EASY
+            type=Dwarvish Explorer
+#else
             type=Dwarvish Pathfinder
+#endif
             id=Molur
             name= _ "Molur"
             {IS_LOYAL}
@@ -561,7 +573,11 @@
 
         [unit]
             side=2
+#ifdef EASY
+            type=Gryphon Master
+#else
             type=Gryphon Rider
+#endif
             id=Elegrild
             name= _ "Elegrild"
             {IS_LOYAL}
@@ -575,7 +591,7 @@
 
         [unit]
             side=2
-            type=Dwarvish Thunderer
+            type={ON_DIFFICULTY (Dwarvish Dragonguard) (Dwarvish Thunderguard) (Dwarvish Thunderer)}
             id=Redbart
             name= _ "Redbart"
             {IS_LOYAL}
@@ -674,7 +690,86 @@
 
         [message]
             speaker=Elynia
-            message= _ "Looking at it in hindsight, there really was no way our infiltration mission wouldn’t turn into a full-fledged assault on the enemy capital. Perhaps it’s better this way, after all. Now, Galas, will you do the honors?"
+            message= _ "Looking at it in hindsight, there really was no way our infiltration mission wouldn’t turn into a full-fledged assault on the enemy capital. Perhaps it’s better this way, after all."
+        [/message]
+
+        # Copying over the Clockwork Dragon from Ashen Hearts to here until I can figure out how to reload the explosive bow object:
+#ifdef EASY
+        [message]
+            speaker=Althurin
+            message= _ "Well, if you're going the full-fledged assault route, I might as well let you have this ancient siege engine we found in the mountains. It's from the days back when there were still drakes around for us to fight."
+        [/message]
+        [unit]
+            side=1
+            type=Clockwork Dragon
+            id=Clockwork Dragon
+            name= _ "Clockwork Dragon"
+            {IS_LOYAL}
+            upkeep=free
+            placement=leader
+        [/unit]
+        [message]
+            speaker=Clockwork Dragon
+            message= _ "..."
+        [/message]
+        [message]
+            speaker=Galas
+            message= _ "Thanks!"
+        [/message]
+        # FIXME: this doesn't work:
+#ifdef __UNUSED__
+        [if]
+            [have_unit]
+                [filter]
+                    [has_attack]
+                        name=shrapnel
+                    [/has_attack]
+                [/filter]
+                search_recall_list=yes
+            [/have_unit]
+            [then]
+                [message]
+                    speaker=Galas
+                    message= _ "It looks like we still have that explosive bow with us; let's refill its ammo."
+                [/message]
+                [recall]
+                    [filter]
+                        [has_attack]
+                            name=shrapnel
+                        [/has_attack]
+                    [/filter]
+                [/recall]
+                [store_unit]
+                    [filter]
+                        # I tried leaving the unit on the recall list and using `x,y="recall,recall"` here,
+                        # but that didn't work, either...
+                        [has_attack]
+                            name=shrapnel
+                        [/has_attack]
+                    [/filter]
+                    variable=boomboomunit
+                [/store_unit]
+                [set_variable]
+                    name=boomboomunit.variables.shrapnel.count
+                    value=6
+                [/set_variable]
+                [unstore_unit]
+                    variable=boomboomunit
+                [/unstore_unit]
+            [/then]
+            [else]
+                [message]
+                    speaker=Galas
+                    message= _ "If only we still had that explosive bow with us, as well..."
+                [/message]
+            [/else]
+        [/if]
+#endif
+#endif
+
+        [message]
+            speaker=Elynia
+            message= _ "Now, Galas, will you do the honors?"
         [/message]
 
         [message]
@@ -787,6 +882,10 @@
         name=attack
         [filter]
             race=elf
+            # So he doesn't have 2 lines in a row from a single attack:
+            [not]
+                id=Galas
+            [/not]
         [/filter]
 
         [message]
@@ -866,7 +965,7 @@
             [modifications]
                 {TRAIT_STRONG}
                 {TRAIT_RESILIENT}
-                {BOSS_BOOST 60% 3 1 2 1}
+                {BOSS_BOOST {ON_DIFFICULTY 40% 50% 60%} {ON_DIFFICULTY 1 2 3} 1 {ON_DIFFICULTY 1 2 3} 1}
                 [object]
                     silent=yes
                     [effect]
@@ -896,7 +995,11 @@
         [unit]
             animate=yes
             side=6
+#ifdef EASY
+            type=Chaos Invoker
+#else
             type=Chaos Magus
+#endif
             id=Ellerian
             name= _ "Ellerian"
             x,y=$location.x,$location.y
