--- 21_Innuendo.cfg	2020-10-10 13:11:54.000000000 -0400
+++ /Users/ericgallager/wesnoth_mods/campaigns/Invasion_from_the_Unknown/scenarios/21_Innuendo.cfg	2020-09-17 01:47:24.000000000 -0400
@@ -4,7 +4,7 @@
     id=21_Innuendo
     name= _ "Innuendo"
     {MAP 21_Innuendo.map}
-    {TURNS 98 93 90}
+    {TURNS 99 95 91}
     next_scenario=22A_Face_of_the_Enemy
     victory_when_enemies_defeated=no
 
@@ -29,7 +29,7 @@
         shroud=yes
 
         {NO_ECONOMY}
-        # Minimum gold for 8 recalls
+        # Minimum gold for 8 recalls; keep constant:
         gold=160
 
         # wmllint: recognize Galas
@@ -68,7 +68,7 @@
         [modifications]
             {TRAIT_SLOW}
             {TRAIT_RESILIENT}
-            {BOSS_BOOST 100% 3 1 2 0}
+            {BOSS_BOOST {DIFF 80% 90% 100%} {DIFF 1 2 3} 1 {DIFF 1 2 3} 0}
         [/modifications]
 
         {SIDE_GENERIC_UNIT (Shaxthal Assault Drone) 34 14} {FACING sw}
@@ -91,7 +91,11 @@
             name= _ "Aelyriun-Balal"
             gender=female
             x,y=39,13
+#ifdef EASY
+            upkeep=full
+#else
             upkeep=free
+#endif
             facing=sw
             [modifications]
                 {TRAIT_STRONG}
@@ -107,7 +111,7 @@
         {CHAOS_FLAG}
 
         recruit=Ghoul,Skeleton Archer,Ghost,Vampire Bat,Bone Shooter,Walking Corpse
-        {INCOME 5 6 6}
+        {INCOME 2 4 6}
 
         {ai/aliases/stable_singleplayer.cfg}
         [ai]
@@ -127,9 +131,9 @@
         {SIDE_GENERIC_GUARD (Abomination) 13 11} {FACING se}
         {SIDE_GENERIC_GUARD (Abomination) 10 12} {FACING se}
 
-        {SIDE_GENERIC_GUARD (Ghast) 13 13} {FACING se}
+        {SIDE_GENERIC_GUARD ({DIFF Ghoul Necrophage Ghast}) 13 13} {FACING se}
 
-        {SIDE_GENERIC_GUARD ({DIFF Necrophage Ghast Ghast}) 17 17} {FACING sw}
+        {SIDE_GENERIC_GUARD ({DIFF Ghoul Necrophage Ghast}) 17 17} {FACING sw}
 
         {SIDE_GENERIC_GUARD (Revenant) 16 14} {FACING sw}
         {SIDE_GENERIC_GUARD (Banebow)  14 15} {FACING se}
@@ -148,15 +152,15 @@
         {SIDE_GENERIC_GUARD (Spectre) 51  4} {FACING sw}
         {SIDE_GENERIC_GUARD (Shadow) 59  7} {FACING sw}
 
-        {SIDE_GENERIC_GUARD (Draug) 32 10} {FACING se}
-        {SIDE_GENERIC_GUARD (Draug) 34  9} {FACING sw}
+        {SIDE_GENERIC_GUARD ({DIFF Revenant Deathblade Draug}) 32 10} {FACING se}
+        {SIDE_GENERIC_GUARD ({DIFF Skeleton Revenant Draug}) 34  9} {FACING sw}
 
         {SIDE_GENERIC_GUARD (Wraith)  8  8} {FACING sw}
         {SIDE_GENERIC_GUARD (Wraith) 11  7} {FACING sw}
 
-        {SIDE_GENERIC_GUARD (Spectre) 14 18} {FACING sw}
+        {SIDE_GENERIC_GUARD ({DIFF Shadow Wraith Spectre}) 14 18} {FACING sw}
 
-        {SIDE_GENERIC_GUARD (Spectre) 61 29} {FACING ne}
+        {SIDE_GENERIC_GUARD ({DIFF Ghost Wraith Spectre}) 61 29} {FACING ne}
     [/side]
 
     [side]
@@ -181,7 +185,7 @@
             {TRAIT_BIOMECHANICAL}
             {TRAIT_ARMORED}
             {TRAIT_RESILIENT}
-            {BOSS_BOOST 90% 2 0 1 1}
+            {BOSS_BOOST {DIFF 70% 80% 90%} {DIFF 0 1 2} 0 1 1}
             [object]
                 [effect]
                     apply_to=image_mod
@@ -242,7 +246,7 @@
         {CHAOS_FLAG}
 
         recruit=Chaos Invader,Chaos Headhunter,Chaos Invoker,Demon
-        {INCOME 4 5 6}
+        {INCOME 2 4 6}
 
         canrecruit=yes
         type=Chaos Arbalestier
@@ -280,7 +284,11 @@
             type=Lich
             id=Mal Abbadur
             name= _ "Mal Abbadur"
+#ifdef EASY
+            upkeep=full
+#else
             upkeep=free
+#endif
             ai_special=guardian
             x,y=22,25
             facing=sw
@@ -1200,7 +1208,7 @@
 
         [modify_side]
             side=5
-            {GOLD 130 140 150}
+            {GOLD 100 125 150}
         [/modify_side]
 
         {MOBILIZE_GUARDIANS (
