--- /Users/ericgallager/Downloads/AfterEI/scenarios/28_Solstice.cfg	2020-02-07 07:00:07.000000000 -0500
+++ /Users/ericgallager/wesnoth_mods/campaigns/After_Eastern_Invasion/scenarios/28_Solstice.cfg	2021-09-22 04:38:10.000000000 -0400
@@ -460,6 +460,17 @@
 y= 51, 1, 1,27, 1,10, 1,15#enddef
 #
 #define TUNNEL_TELEPORT_DWARF STARTXY ENDXY
+#ifdef EASY
+#ifdef DEBUG_MODE
+  [event]
+    name=prestart
+    [label]
+      {STARTXY}
+      text="to {ENDXY}"
+    [/label]
+  [/event]
+#endif
+#endif
   [event]
     name=moveto
     first_time_only=no
@@ -496,6 +507,17 @@
 #enddef
 #
 #define TUNNEL_TELEPORT_ALL STARTXY ENDXY
+#ifdef EASY
+#ifdef DEBUG_MODE
+  [event]
+    name=prestart
+    [label]
+      {STARTXY}
+      text="to {ENDXY}"
+    [/label]
+  [/event]
+#endif
+#endif
   [event]
     name=moveto
     first_time_only=no
@@ -652,7 +674,7 @@
   name=_"Solstice"
   next_scenario=29_Epilogue
   map_data="{~add-ons/AfterEI/maps/28_Solstice.map}"
-  turns=-1
+  turns=-1 # i.e. unlimited
   victory_when_enemies_defeated=no
 
   {INTRO_AND_SCENARIO_MUSIC traveling_minstrels.ogg into_the_shadows.ogg}
@@ -705,6 +727,7 @@
   [side]
 {FLYNN_LF}
     gold={MIN_START_GOLD_ELGENEFAR}	# *Elgenefar* starting gold (or carryover, if higher).
+    {INCOME 3 2 1}
     shroud=yes
   [/side]
 
@@ -758,7 +781,9 @@
     flag=flags/ragged-flag-[1~6].png:150
     flag_icon=flags/ragged-flag-icon.png
     {GOLD 400 500 600}
-#    {INCOME 25 30 35}					# Instead of extra income, free units are spawned
+#ifdef NO_FREE_UNIT_SPAWNS
+    {INCOME 25 30 35} # ifdef-ed out because, instead of extra income, free units are spawned
+#endif
     [unit]			# Shaman (controls serpents)
       type=aei_oshe
       id=serpent_controller
@@ -832,7 +857,9 @@
     flag=flags/ragged-flag-[1~6].png:150
     flag_icon=flags/ragged-flag-icon.png
     {GOLD 250 300 350}
-#    {INCOME 20 25 30}		# Instead of extra income, free units are spawned
+#ifdef NO_FREE_UNIT_SPAWNS
+    {INCOME 20 25 30} # ifdef-ed out because, instead of extra income, free units are spawned
+#endif
   [/side]
 
 # S leader
@@ -871,7 +898,9 @@
     flag=flags/ragged-flag-[1~6].png:150
     flag_icon=flags/ragged-flag-icon.png
     {GOLD 250 300 350}
-#    {INCOME 20 25 30}					# Instead of extra income, free units are spawned
+#ifdef NO_FREE_UNIT_SPAWNS
+    {INCOME 20 25 30} # ifdef-ed out because, instead of extra income, free units are spawned
+#endif
   [/side]
 
 # SE leader
@@ -910,7 +939,9 @@
     flag=flags/ragged-flag-[1~6].png:150
     flag_icon=flags/ragged-flag-icon.png
     {GOLD 300 350 400}	# Bit extra to offset spiders
-#    {INCOME 27 30 33}	# Bit extra to offset spiders
+#ifdef NO_FREE_UNIT_SPAWNS
+    {INCOME 27 30 33}	# Bit extra to offset spiders
+#endif
   [/side]
 
   [side]
@@ -987,7 +1018,7 @@
       [/avoid]
     [/ai]
     color=gold
-    gold=300
+    {GOLD 280 290 300}
   [/side]
 
 # Controlled serpents
@@ -1104,8 +1135,10 @@
 # Regeneration (green) ring pickup and drop logic
 {AEI_PICKDROP_RING_REGEN}
 
-#    {AEI_PICKDROP_TEST_IF_UNAVAILABLE ring_02b}  --- not needed
-#    {AEI_PICKDROP_TEST_IF_UNAVAILABLE amulet_03} --- not needed
+#ifdef __UNUSED__
+    {AEI_PICKDROP_TEST_IF_UNAVAILABLE ring_02b}  # --- not needed
+    {AEI_PICKDROP_TEST_IF_UNAVAILABLE amulet_03} # --- not needed
+#endif
 
 
 # Race-dependent dialogue
@@ -1617,7 +1650,7 @@
         [set_variables]
           name=rspeak
           [value]
-            d="It's nae guid tae me."
+            d="This axe is nae guid tae me; a dwarf more skilled with axes would be a better wielder of it."
             e="This weapon is unsuitable: it is heavy, unrefined and inelegant "+{ASIDE _"just like its makers..."}
             h="Hmm, well-made but balanced for dwarves.  No use to me."
             l="Uss Ssaurianss sshun axess."
@@ -1712,7 +1745,7 @@
         [set_variables]
           name=rspeak
           [value]
-            d="Ah cannae use this!"
+            d="Ah cannae use this Thunderdust!"
             e="Dwarves may be happy to lug explosives around.  I am not!"
             h="Do I look like a dwarf?"
             l="Thiss dusst is unsstable, unssafe and thuss usselessss to uss ssaurianss."
@@ -2011,6 +2044,11 @@
     [/filter]
     [cancel_action]		# Interrupt movement if necessary
     [/cancel_action]
+    [message]
+      # TODO: make this race-dependent dialogue:
+      speaker=unit
+      message=_"Aieee! These stepping stones are crumbling!"
+    [/message]
     [sound]
       name=water-blast.wav
     [/sound]
@@ -2106,24 +2144,26 @@
 #
 # *** Superceded: now use return guardian micro_ai ***
 #
+#ifdef FREEZE_WOLVES_MANUALLY
 # Alpha and beta wolves stay put until Aknzlay killed
-#  [event]
-#    name=side 5 turn refresh
-#    first_time_only=no
-#    [filter_condition]
-#      [have_unit]
-#        id=Aknzlay,Znottiyanki
-#        count=2
-#      [/have_unit]
-#    [/filter_condition]
-#    [modify_unit]
-#      [filter]
-#        id=side5_alpha,side5_beta
-#      [/filter]
-#      moves=0
-#    [/modify_unit]
-#  [/event]
-#
+  [event]
+    name=side 5 turn refresh
+    first_time_only=no
+    [filter_condition]
+      [have_unit]
+        id=Aknzlay,Znottiyanki
+        count=2
+      [/have_unit]
+    [/filter_condition]
+    [modify_unit]
+      [filter]
+        id=side5_alpha,side5_beta
+      [/filter]
+      moves=0
+    [/modify_unit]
+  [/event]
+#endif
+
   [event]		# First time dialogue for recruit fail
     name=rfw_ftd
     [if]
@@ -3883,7 +3923,13 @@
     {CLEAR_VARIABLE Arrah_save}
     [set_extra_recruit]
       id=Arrah
-      extra_recruit="Cavalryman,Bowman,Longbowman,Spearman,Javelineer,Pikeman,Swordsman"
+#ifdef EASY
+      # A few more generally-useful recruits available on EASY:
+      extra_recruit="Cavalryman,Dragoon,Bowman,Longbowman,Spearman,Javelineer,Pikeman,Swordsman,Fencer,Heavy Infantryman"
+#else
+      # OK to use raw peasants instead of AEI_Peasants here because their only change is in recall cost, and this is the last scenario, so it doesn't matter:
+      extra_recruit="Cavalryman,Bowman,Longbowman,Spearman,Javelineer,Pikeman,Swordsman,Peasant,Woodsman,Sergeant"
+#endif
     [/set_extra_recruit]
     [terrain]
       {ARRAHXY}
@@ -4107,7 +4153,7 @@
     [fire_event]
       name=gold_set_or_add
     [/fire_event]
-   {CLEAR_VARIABLE AEI_Elgenefar_gold_save}
+    {CLEAR_VARIABLE AEI_Elgenefar_gold_save}
     [message]
       speaker=Elgenefar
       message= _ "Gallant Earl Flynn, greetings and salutations!  We will cleanse the forest of orcish stain and stench!"
@@ -4291,7 +4337,7 @@
       [/sound]
       [gold]
         side=1
-        amount=150
+        amount={ON_DIFFICULTY 190 170 150}
       [/gold]
       [if]		# Dialogue varies according to whether Flynn has any dwarf runesmiths or advancements
         [have_unit]
@@ -5016,10 +5062,10 @@
       name=perf_hdr_text,hidden_treasures_found,ht_02b_text,real_chest_text,bbrecruit_text,ht_02b,hidden_treasures_rating
     [/clear_variable]
 #
-    [message]			# Add rock music (optional)
+    [message]			# Add rock/metal music (optional)
       speaker=narrator
       image=misc/music.png
-      message= _ "Would you like to include a few rock tunes in the playlist?"
+      message= _ "Would you like to include a few rock/metal tunes in the playlist?"
       [option]
         label= _ "Yes, why not?"
         [command]
@@ -5227,7 +5273,8 @@
       speaker=m_d
       message = _ "Ah got here usin' yon crawlway a short step nor-west o' the ford behind me."
     [/message]
-    {AEI_HIGHLIGHT_IMAGE 6 51 items/gohere.png scenery/cave-entrance.png 150 3}		# = MD_SOUTHXY
+    # Vary delay and number of blinks here to make players on easier difficulties pay more attention:
+    {AEI_HIGHLIGHT_IMAGE 6 51 items/gohere.png scenery/cave-entrance.png {ON_DIFFICULTY 190 170 150} {ON_DIFFICULTY 5 4 3}} # = MD_SOUTHXY
 #
 # Antidote potion-dependent dialogue (dwarf)
     [set_variable]
@@ -5316,7 +5363,7 @@
  
 I heard what, ah, our dwarven friend reported and I bear similar news: Elgenefar's forces draw near.  Just as the orcs are, for now, disregarding the mountains they similarly pay scant heed to the forests.  Before you ask, $ranger_dialogue_conditional
  
-I passed a secluded area largely hidden by palm forest and mountains east of here.  Mayhap it will serve as the ""small defensible region"" your plan requires?  Also, I was greatly fortunate to find stepping stones across the river, presumably revealed by the low summer waters.  They are treacherous and unstable so whilst us lightfoot elves can, with care, cross safely I would <b><span foreground='#ffff00'>strongly</span></b> advise you avoid them.
+I passed a secluded area largely hidden by palm forest and mountains east of here.  Mayhap it will serve as the ""small defensible region"" your plan requires?  Also, I was greatly fortunate to find stepping stones across the river, presumably revealed by the low summer waters.  They are treacherous and unstable, so whilst us lightfoot elves can, with care, cross safely, <u>I would <b><span foreground='#ffff00'>strongly</span></b> advise you <i>avoid</i> trying to cross them with anyone else.</u>
  
 Finally, do you have any message for Elgenefar ere I depart with word of your arrival?"
     [/message]
@@ -5797,6 +5844,17 @@
       id=Scampa
       x,y=54,57
     [/recall]
+#ifdef EASY
+    [terrain]
+      x,y=53,59
+      terrain=Gg^Vct
+    [/terrain]
+    [capture_village]
+      x,y=53,59
+      side=1
+    [/capture_village]
+#endif
+          
 #
 # Remove unpoisonable status (may not be any)
 # Note that [object] [filter] only applies to
@@ -6082,7 +6140,7 @@
         [modifications]
           {TRAIT_RESILIENT}
           {TRAIT_STRONG}
-          {PACK_LEADER 15 2}
+          {PACK_LEADER {ON_DIFFICULTY 9 12 15} {ON_DIFFICULTY 1 2 3}}
         [/modifications]
         generate_name=no
         random_traits=no
@@ -6099,7 +6157,7 @@
         [modifications]
           {TRAIT_RESILIENT}
           {TRAIT_STRONG}
-          {PACK_LEADER 10 2}
+          {PACK_LEADER {ON_DIFFICULTY 8 9 10} {ON_DIFFICULTY 0 1 2}}
         [/modifications]
         generate_name=no
         random_traits=no
